---
title: "Spatial and temporal variation"
output:
  blogdown::html_page:
    toc: true
weight: 1
---


<div id="TOC">
<ul>
<li><a href="#gepia" id="toc-gepia">5.1.1 <strong>GEPIA</strong></a></li>
<li><a href="#humanbase-expecto" id="toc-humanbase-expecto">5.1.2 <strong>HumanBase-ExPecto</strong></a></li>
<li><a href="#expression-atlas" id="toc-expression-atlas">5.1.3 <strong>Expression Atlas</strong></a></li>
<li><a href="#toppcluster" id="toc-toppcluster">5.1.4 <strong>ToppCluster</strong></a></li>
<li><a href="#gent2" id="toc-gent2">5.1.5 <strong>GENT2</strong></a></li>
<li><a href="#tissues" id="toc-tissues">5.1.6 <strong>TISSUES</strong></a></li>
<li><a href="#tissueenrich" id="toc-tissueenrich">5.1.7 <strong>TissueEnrich</strong></a></li>
<li><a href="#organizer" id="toc-organizer">5.1.8 <strong>ORGANizer</strong></a></li>
<li><a href="#dets" id="toc-dets">5.1.9 <strong>deTS</strong></a></li>
<li><a href="#webcsea" id="toc-webcsea">5.1.10 <strong>WebCSEA</strong></a></li>
<li><a href="#tengexa" id="toc-tengexa">5.1.11 <strong>TEnGExA</strong></a></li>
<li><a href="#dynamic-bm" id="toc-dynamic-bm">5.1.12 <strong>Dynamic-BM</strong></a></li>
<li><a href="#adeip" id="toc-adeip">5.1.13 <strong>ADEIP</strong></a></li>
</ul>
</div>

<hr />
<div id="gepia" class="section level2">
<h2>5.1.1 <strong>GEPIA</strong></h2>
<p><strong>Introduction</strong>: GEPIA is a newly developed interactive web server for analyzing the RNA sequencing expression data of 9,736 tumors and 8,587 normal samples from the TCGA and the GTEx projects, using a standard processing pipeline. GEPIA provides customizable functions such as tumor/normal differential expression analysis, profiling according to cancer types or pathological stages, patient survival analysis, similar gene detection, correlation analysis and dimensionality reduction analysis. Meanwhile, the GEPIA2021, a standalone extension with multiple deconvolution-based analysis for GEPIA. We deconvolute each sample tool in TCGA/GTEx with the bioinformatics tools CIBERSORT, EPIC and quanTIseq.
<img src="/images/GEPIA.jpg?width=50pc" alt="GEPIA" /></p>
<p><strong>Web server</strong>: <a href="http://gepia.cancer-pku.cn/index.html">GEPIA</a></p>
<hr />
</div>
<div id="humanbase-expecto" class="section level2">
<h2>5.1.2 <strong>HumanBase-ExPecto</strong></h2>
<p><strong>Introduction</strong>: ExPecto is a framework for ab initio sequence-based prediction of mutation gene expression effects and disease risks. With this web interface, we provide an explorer of tissue-specific expression effect predictions. The current release contains all single nucleotide substitutions within 1kb to the representative TSS of a gene and all 1000 Genomes variants that passed a minimum predicted effect threshold (&gt;0.3 log fold-change in any tissue). ExPecto uses exponential basis function-based linear models upon deep convolutional network model of chromatin effects. ExPecto predicts expression levels directly from sequence and is capable of predicting effects of sequence variations.
<img src="/images/ExPecto2.jpg?width=50pc" alt="ExPecto" /></p>
<p><strong>Web server</strong>: <a href="http://gepia.cancer-pku.cn/index.html">ExPecto</a>
<img src="/images/ExPecto.jpg?width=50pc" alt="ExPecto2" /></p>
<hr />
</div>
<div id="expression-atlas" class="section level2">
<h2>5.1.3 <strong>Expression Atlas</strong></h2>
<p><strong>Introduction</strong>: Expression Atlas is an open science resource that gives users a powerful way to find information about gene and protein expression. Expression Atlas provide the scientific community with freely available information on the abundance and localisation of RNA (and proteins) across species and biological conditions such as different tissues, cell types, developmental stages and diseases among others.
<img src="/images/ExpressionAtlas2.png?width=50pc" alt="ExpressionAtlas" /></p>
<p><strong>Web server</strong>: <a href="http://www.ebi.ac.uk/gxa">Expression Atlas</a>
<img src="/images/ExpressionAtlas.jpg?width=50pc" alt="ExpressionAtlas2" /></p>
<hr />
</div>
<div id="toppcluster" class="section level2">
<h2>5.1.4 <strong>ToppCluster</strong></h2>
<p><strong>Introduction</strong>: ToppCluster is a tool for performing multi-cluster gene functional enrichment analyses on large scale data (microarray experiments with many time-points, cell-types, tissue-types, etc.). ToppCluster facilitates co-analysis of multiple gene lists and yields as output a rich functional map showing the shared and list-specific functional features. The output can be visualized in tabular, heatmap or network formats using built-in options as well as third-party software. ToppCluster uses the hypergeometric test to obtain functional enrichment achieved via the gene list enrichment analysis option available in ToppGene</p>
<p><strong>Web server</strong>: <a href="http://toppcluster.cchmc.org">ToppCluster</a>
<img src="/images/ToppCluster.jpg?width=50pc" alt="ToppCluster" /></p>
<hr />
</div>
<div id="gent2" class="section level2">
<h2>5.1.5 <strong>GENT2</strong></h2>
<p><strong>Introduction</strong>: Gene Expression database of Normal and Tumor tissues 2 (GENT2) is an updated version of GENT, which has provided a user-friendly search platform for gene expression patterns across different normal and tumor tissues compiled from public gene expression data sets. We refactored GENT2 with recent technologies such as Apache Lucene indexing for fast search and Google Web Toolkit (GWT) framework for a user-friendly web interface. Now, GENT2 contains more than 68,000 samples and has several new useful functions. First, GENT2 now provides gene expression across 72 different tissues compared to 57 in GENT. Second, with increasing importance of tumor subtypes, GENT2 provides an option to study the differential expression and its prognostic significance based on tumor subtypes. Third, whenever available, GENT2 provides prognostic information of a gene of interest. Fourth, GENT2 provides a meta-analysis of survival information to provide users more reliable prognostic value of a gene of interest.</p>
<p><strong>Web server</strong>: <a href="http://gent2.appex.kr">GENT2</a>
<img src="/images/GENT2.jpg?width=60pc" alt="GENT2" /></p>
<hr />
</div>
<div id="tissues" class="section level2">
<h2>5.1.6 <strong>TISSUES</strong></h2>
<p><strong>Introduction</strong>: TISSUES is a weekly updated web resource (including four species, human, mouse, rat and pig) that integrates evidence on tissue expression from manually curated literature, proteomics and transcriptomics screens, and automatic text mining. TISSUES map all evidence to common protein identifiers and Brenda Tissue Ontology terms, and further unify it by assigning confidence scores that facilitate comparison of the different types and sources of evidence.</p>
<p><strong>Web server</strong>: <a href="https://tissues.jensenlab.org/Search">TISSUES</a></p>
<hr />
</div>
<div id="tissueenrich" class="section level2">
<h2>5.1.7 <strong>TissueEnrich</strong></h2>
<p><strong>Introduction</strong>: R package for tissue-specific gene enrichment. TissueEnrich defines tissue-specific genes using RNA-Seq data from the HPA, GTEx, and mouse ENCODE. In order to make the tissue-specific gene calculations more robust, we only used tissues that had ≥2 biological replicates. TissueEnrich enables users to provide the background genes for carrying out tissue-specific gene enrichment. In this case, instead of using all the genes in the dataset, a background gene set is being used to carry out the enrichment analysis. It should be noted that the background gene set must have all the genes of the input gene set.
<strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p>
<pre class="r"><code>BiocManager::install(&quot;TissueEnrich&quot;)</code></pre>
<p>TissueEnrich R package can be easily installed from Github using devtools <a href="https://github.com/Tuteja-Lab/TissueEnrich" class="uri">https://github.com/Tuteja-Lab/TissueEnrich</a>:</p>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
library(devtools)
install_github(&quot;Tuteja-Lab/TissueEnrich&quot;)</code></pre>
<p><strong>Application</strong>: More examples can download from <a href="https://bioconductor.org/packages/3.7/bioc/vignettes/TissueEnrich/inst/doc/TissueEnrich.html">here</a>.</p>
<p><strong>Web server</strong>: <a href="http://tissueenrich.gdcb.iastate.edu/">TissueEnrich</a>
<img src="/images/TissueEnrich.jpeg?width=50pc" alt="TissueEnrich" /></p>
<hr />
</div>
<div id="organizer" class="section level2">
<h2>5.1.8 <strong>ORGANizer</strong></h2>
<p><strong>Introduction</strong>: Gene ORGANizer is a tool designed to allow users to analyze the relationships between genes and organs. The tool contains annotations for over 7,000 genes, which are linked to ~150 body parts, divided into 4 levels of hierarchy: organs (e.g., stomach), systems (e.g., digestive), regions (e.g., abdomen) and germ layers (e.g., endoderm). Users can either Browse a specific gene to see which body parts the gene affects, or enter a list of genes in ORGANize to find out what body parts are enriched or depleted within the list. Results are displayed in an interactive user-friendly body map visualization and in a summary table.</p>
<p><strong>Web server</strong>: <a href="http://geneorganizer.huji.ac.il/">ORGANizer</a>
<img src="/images/ORGANizer.jpg?width=60pc" alt="ORGANizer" />
<img src="/images/ORGANizer2.png?width=60pc" alt="ORGANizer2" /></p>
<hr />
</div>
<div id="dets" class="section level2">
<h2>5.1.9 <strong>deTS</strong></h2>
<p><strong>Introduction</strong>: An R package conduct tissue-specific enrichment analysis with two built-in reference panels. Statistical methods are developed and implemented for detecting tissue-specific genes and for enrichment test of different forms of query data</p>
<p><strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p>
<pre class="r"><code>install.packages(&quot;deTS&quot;)</code></pre>
<p>deTS R package can be easily installed from Github using devtools <a href="https://github.com/bsml320/deTS" class="uri">https://github.com/bsml320/deTS</a>:</p>
<pre class="bash"><code>git clone https://github.com/bsml320/deTS.git

cd deTS</code></pre>
<p>Then open the R:</p>
<pre class="r"><code>install.packages(&quot;deTS_1.0.tar.gz&quot;)</code></pre>
<p><strong>Application</strong>: More examples can download from <a href="https://github.com/bsml320/deTS">here</a>.</p>
<hr />
</div>
<div id="webcsea" class="section level2">
<h2>5.1.10 <strong>WebCSEA</strong></h2>
<p><strong>Introduction</strong>: WebCSEA (Web-based Cell-type Specific Enrichment Analysis of Genes) provides a gene set query among a systematic collection of tissue-cell-type expression signatures. For each query, we will generate cell-type-specific enrichment analysis (CSEA) raw p-value, combined p-value by permutation-based method, genes shared between queried gene list and tissue-cell-type signatures, heatmap that interactively displays the cell-type specificity across 1,355 human tissue-cell types (TCs), summary of cell-type specificity by human organ system and top-ranked tissues and general cell types. Users can also filter, prioritize, and compare the CSEA results from cell types from one or multiple tissues. Users can further download all the results of CSEA and visualize them. Overall, WebCSEA provides a comprehensive map of major human TCs and uses a permutation-based method to overcome the bias raised from different lengths of signature genes among TCs as well as the length of query genes.</p>
<p>WebCSEA team manually curated more than 5.5 million cells from 111 human tissues panels and 1,355 TCs from 61 adult and fetal tissues across 12 human organ systems (11 organ systems + sensory system). We used our t-statistic method to obtain the tissue-cell-type signatures panel by tissue. To construct a biological meaningful gene set for permutation shuffle, we curated ~5,600 genome-wide association studies (GWAS) and ~3,700 rare-variants association phenotypes from UK BioBank, GWAS catalog, and Genebass for trait-associated gene sets (TAGs). After stringent quality control criteria, we permutated the null model as a matrix of 19,663 TAGs by 1,355 TCs. For each inquiry gene list, we conduct CSEA across 1,355 TCs and get the permutated p-values by ranking among 19,663 TAGs and 1,355 TCs. Then we use Fisher’s method to combine an overall p-value from the permutated p-values by TAGs and by TCs.
<img src="/images/WebCSEA.png?width=60pc" alt="WebCSEA" /></p>
<p><strong>Application</strong>: More examples can download from <a href="https://bioinfo.uth.edu/webcsea/tutorial.php?csrt=8124369579383014372">here</a>.
<strong>Web server</strong>: <a href="https://bioinfo.uth.edu/webcsea/">WebCSEA</a></p>
<hr />
</div>
<div id="tengexa" class="section level2">
<h2>5.1.11 <strong>TEnGExA</strong></h2>
<p><strong>Introduction</strong>: R package has been developed to perform tissue-enrichment analysis of any large number of genes with any number of tissues, irrespective of any organism provided only the read count matrix.</p>
<p><strong>Installation</strong>: TEnGExA R package can be easily installed from Github using devtools <a href="https://github.com/ubagithub/TEnGExA" class="uri">https://github.com/ubagithub/TEnGExA</a>:</p>
<pre class="r"><code>install.packages(&quot;devtools&quot;)

library(devtools)

install_github(&quot;ubagithub/TEnGExA&quot;)</code></pre>
<p><strong>Application</strong>: More examples can download from <a href="https://github.com/ubagithub/TEnGExA">here</a>.</p>
<hr />
</div>
<div id="dynamic-bm" class="section level2">
<h2>5.1.12 <strong>Dynamic-BM</strong></h2>
<p><strong>Introduction</strong>: Dynamic-BM stands for Dynamic BodyMap, which provides massive and comprehensive dynamic RNA-seq data archives and analyses. The current version of Dynamic-BM contains 2,203 RNA-seq samples (114 datasets) of more than 20 tissues in 5 species. The main sections of this website are Home, Expression, Datasets, Dev-lncRNAs, Analysis and Download. The Navigation bar on the right provides a quick access to the help file of each section.
<img src="/images/DynamicBM.png?width=50pc" alt="DynamicBM" /></p>
<p><strong>Application</strong>: More examples can download from <a href="http://bioinfo.ibp.ac.cn/Dynamic-BM/help.php">here</a>.
<strong>Web server</strong>: <a href="http://bioinfo.ibp.ac.cn/Dynamic-BM/index.php">Dynamic-BM</a>
<img src="/images/DynamicBM2.png?width=60pc" alt="DynamicBM2" /></p>
<hr />
</div>
<div id="adeip" class="section level2">
<h2>5.1.13 <strong>ADEIP</strong></h2>
<p><strong>Introduction</strong>: ADEIP (An integrating platform of age-dependent expression and immune profiles across human tissues) provided the differences of gene expression and cell ratios among different ages and genders. ADEIP includes data on a much larger scale (16704 samples, 54592 genes, 30 tissues, 22 cell types, 34 types of immune function, 2 gender) than previous databases.
<img src="/images/ADEIP.png?width=60pc" alt="ADEIP" /></p>
<p><strong>Application</strong>: More examples can download from <a href="http://gb.whu.edu.cn/ADEIP/tutorials">here</a>.</p>
<p><strong>Web server</strong>: <a href="http://gb.whu.edu.cn/ADEIP/">ADEIP</a></p>
</div>
