---
title: "Spatial and temporal variation"
output:
  blogdown::html_page:
    toc: true
weight: 3
---

**Breaking**: 

---

## 9.1.1 **GEPIA**

- **Introduction**: GEPIA is a newly developed interactive web server for analyzing the RNA sequencing expression data of 9,736 tumors and 8,587 normal samples from the TCGA and the GTEx projects, using a standard processing pipeline. GEPIA provides customizable functions such as tumor/normal differential expression analysis, profiling according to cancer types or pathological stages, patient survival analysis, similar gene detection, correlation analysis and dimensionality reduction analysis. Meanwhile, the GEPIA2021, a standalone extension with multiple deconvolution-based analysis for GEPIA. We deconvolute each sample tool in TCGA/GTEx with the bioinformatics tools CIBERSORT, EPIC and quanTIseq. 
![GEPIA](/images/GEPIA.jpg?width=50pc)

- **Web server**: [GEPIA](http://gepia.cancer-pku.cn/index.html)

---

## 9.1.2 **HumanBase-ExPecto**

![](/images/ExPecto.jpg?width=50pc)

- **Introduction**: ExPecto is a framework for ab initio sequence-based prediction of mutation gene expression effects and disease risks. With this web interface, we provide an explorer of tissue-specific expression effect predictions. The current release contains all single nucleotide substitutions within 1kb to the representative TSS of a gene and all 1000 Genomes variants that passed a minimum predicted effect threshold (>0.3 log fold-change in any tissue). ExPecto uses exponential basis function-based linear models upon deep convolutional network model of chromatin effects. ExPecto predicts expression levels directly from sequence and is capable of predicting effects of sequence variations.

- **Web server**: [ExPecto](http://gepia.cancer-pku.cn/index.html)
![ExPecto2](/images/ExPecto2.jpg?width=50pc)

---

## 9.1.3 **Expression Atlas**

- **Introduction**: Expression Atlas is an open science resource that gives users a powerful way to find information about gene and protein expression. Expression Atlas provide the scientific community with freely available information on the abundance and localisation of RNA (and proteins) across species and biological conditions such as different tissues, cell types, developmental stages and diseases among others.
![ExpressionAtlas](/images/ExpressionAtlas.jpg?width=50pc)

- **Web server**: [Expression Atlas](http://www.ebi.ac.uk/gxa)
![ExpressionAtlas2](/images/ExpressionAtlas2.png?width=50pc)

---

## 9.1.4 **ToppCluster**

- **Introduction**:  ToppCluster is a tool for performing multi-cluster gene functional enrichment analyses on large scale data (microarray experiments with many time-points, cell-types, tissue-types, etc.). ToppCluster facilitates co-analysis of multiple gene lists and yields as output a rich functional map showing the shared and list-specific functional features. The output can be visualized in tabular, heatmap or network formats using built-in options as well as third-party software. ToppCluster uses the hypergeometric test to obtain functional enrichment achieved via the gene list enrichment analysis option available in ToppGene

- **Web server**: [ToppCluster](http://toppcluster.cchmc.org)
![ToppCluster](/images/ToppCluster.jpg?width=50pc)

---

## 9.1.5 **TISSUES**

- **Introduction**:  TISSUES is a weekly updated web resource (including four species, human, mouse, rat and pig) that integrates evidence on tissue expression from manually curated literature, proteomics and transcriptomics screens, and automatic text mining. TISSUES map all evidence to common protein identifiers and Brenda Tissue Ontology terms, and further unify it by assigning confidence scores that facilitate comparison of the different types and sources of evidence.

- **Web server**: [TISSUES](https://tissues.jensenlab.org/Search)

---
