---
title: "Dynamics inference"
output:
  blogdown::html_page:
    toc: true
weight: 2
---


<div id="TOC">
<ul>
<li><a href="#slingshot" id="toc-slingshot">6.2.2.1 <strong>slingshot</strong></a></li>
<li><a href="#cellrank" id="toc-cellrank">6.2.2.2 <strong>CellRank</strong></a></li>
<li><a href="#cytotree" id="toc-cytotree">6.2.2.3 <strong>CytoTree</strong></a></li>
<li><a href="#veloviz" id="toc-veloviz">6.2.2.4 <strong>VeloViz</strong></a></li>
<li><a href="#asc-seurat" id="toc-asc-seurat">6.2.2.5 <strong>Asc-Seurat</strong></a></li>
<li><a href="#vetra" id="toc-vetra">6.2.2.6 <strong>VeTra</strong></a></li>
<li><a href="#timeor" id="toc-timeor">6.2.2.7 <strong>TIMEOR</strong></a></li>
<li><a href="#unitvelo" id="toc-unitvelo">6.2.2.8 <strong>UniTVelo</strong></a></li>
<li><a href="#tips" id="toc-tips">6.2.2.9 <strong>TIPS</strong></a></li>
<li><a href="#velo-predictor" id="toc-velo-predictor">6.2.2.10 <strong>Velo-Predictor</strong></a></li>
<li><a href="#origins" id="toc-origins">6.2.2.11 <strong>ORIGINS</strong></a></li>
<li><a href="#paga" id="toc-paga">6.2.2.12 <strong>PAGA</strong></a></li>
<li><a href="#scorpius" id="toc-scorpius">6.2.2.13 <strong>SCORPIUS</strong></a></li>
<li><a href="#stemnet" id="toc-stemnet">6.2.2.14 <strong>STEMNET</strong></a></li>
<li><a href="#monocle3" id="toc-monocle3">6.2.2.15 <strong>Monocle3</strong></a></li>
<li><a href="#scvelo" id="toc-scvelo">6.2.2.16 <strong>scVelo</strong></a></li>
<li><a href="#velovi" id="toc-velovi">6.2.2.17 <strong>veloVI</strong></a></li>
</ul>
</div>

<p>Trajectory inference aims at estimating dynamic changes in a single-cell transcriptome landscape, assuming that the cell-wise transcriptome is a static snapshot at a time point along some cellular process. The pseudotime and RNA velocity both are considered as the trajectory inference.Highly variable genes, represented by DEGs, thus contribute an important role in the pseudotime and RNA velocity analysis.</p>
<p><strong>Breaking</strong>: A plethora of trajectory inference methods has been developed over the past few years, and 11 user-friendly tools are recommended in this document.</p>
<hr />
<div id="slingshot" class="section level2">
<h2>6.2.2.1 <strong>slingshot</strong></h2>
<p><img src="/images/slingshot_sticker.png?width=20pc" /></p>
<p><strong>Introduction</strong>: Provides functions for inferring continuous, branching lineage structures in low-dimensional data. Slingshot was designed to model developmental trajectories in single-cell RNA sequencing data and serve as a component in an analysis pipeline after dimensionality reduction and clustering. It is flexible enough to handle arbitrarily many branching events and allows for the incorporation of prior knowledge through supervised graph construction.</p>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly=TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;kstreet13/slingshot&quot;)</code></pre>
<p>Python Version: An implementation of the Slingshot algorithm in python can be found here: <a href="https://github.com/mossjacob/pyslingshot" class="uri">https://github.com/mossjacob/pyslingshot</a></p>
<hr />
</div>
<div id="cellrank" class="section level2">
<h2>6.2.2.2 <strong>CellRank</strong></h2>
<p><strong>Introduction</strong>: CellRank is a toolkit to uncover cellular dynamics based on Markov state modeling of single-cell data. It contains two main modules: kernels compute cell-cell transition probabilities and estimators generate hypothesis based on these.</p>
<p><strong>Installation</strong>: CellRank requires Python version &gt;= 3.7 to run. We recommend using Miniconda to manage the environments.</p>
<p>CellRank can be installed via:</p>
<pre class="python"><code>conda install -c conda-forge -c bioconda cellrank
# or with extra libraries, useful for large datasets
conda install -c conda-forge -c bioconda cellrank-krylov</code></pre>
<p><strong>Web server</strong>: <a href="https://cellrank.org/">CellRank</a>.
<img src="/images/CellRank.jpg?width=50pc" alt="CellRank" /></p>
<hr />
</div>
<div id="cytotree" class="section level2">
<h2>6.2.2.3 <strong>CytoTree</strong></h2>
<p><img src="/images/CytoTree.png?width=15pc" /></p>
<p><strong>Introduction</strong>: CytoTree, an R/Bioconductor package for trajectory inference, pseudotime estimation and visualization of flow and mass cytometry data. The CytoTree package offers a complete up-to-date analysis workflow for flow and mass cytometry data that includes subpopulation classification, dimensionality reduction, trajectory construction, differentially expressed marker calculation, pseudotime estimation, intermediate state identification and visualization. The CytoTree runs on several platforms, such as UNIX, Windows, and macOS, and provides an up-to-date, feature-rich and readily scalable workflow.</p>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code># If not already installed
install.packages(&quot;devtools&quot;) 
devtools::install_github(&quot;JhuangLab/CytoTree&quot;)</code></pre>
<p>CytoTree is also available on Bioconductor:</p>
<pre class="r"><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;CytoTree&quot;)</code></pre>
<p>CytoTree Tutorial is provided in <a href="https://ytdai.github.io/CytoTree/index.html" class="uri">https://ytdai.github.io/CytoTree/index.html</a></p>
<hr />
</div>
<div id="veloviz" class="section level2">
<h2>6.2.2.4 <strong>VeloViz</strong></h2>
<p><img src="/images/VeloViz.png?width=15pc" /></p>
<p><strong>Introduction</strong>: VeloViz creates an RNA-velocity-informed 2D embedding for single cell transcriptomics data.</p>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code>require(remotes)
remotes::install_github(&#39;JEFworks-Lab/veloviz&#39;)</code></pre>
<p>VeloViz is also available on Bioconductor:</p>
<pre class="r"><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;veloviz&quot;)</code></pre>
<p>A example is provided in <a href="https://github.com/JEFworks-Lab/veloviz" class="uri">https://github.com/JEFworks-Lab/veloviz</a></p>
<hr />
</div>
<div id="asc-seurat" class="section level2">
<h2>6.2.2.5 <strong>Asc-Seurat</strong></h2>
<p><strong>Introduction</strong>: Asc-Seurat (Analytical single-cell Seurat-based web application) is a web application based on Shiny. Pronounced as “ask Seurat”, it provides a click-based, easy-to-install, and easy-to-use interface that allows the execution of all steps necessary for scRNA-seq analysis. It integrates many of the capabilities of the Seurat and Dynverse and also allows an instantaneous functional annotation of genes of interest using BioMart.</p>
<p><strong>Installation</strong>: To install Asc-Seurat, it is necessary to have Docker installed on the machine. Docker needs to be correctly installed and configured in the user’s machine. Check the installation instructions provided by Docker at (<a href="https://docs.docker.com/engine/install" class="uri">https://docs.docker.com/engine/install</a>).</p>
<p><strong>Web server</strong>: <a href="https://asc-seurat.readthedocs.io/en/latest/index.html">Asc-Seurat</a>.</p>
<hr />
</div>
<div id="vetra" class="section level2">
<h2>6.2.2.6 <strong>VeTra</strong></h2>
<p><strong>Introduction</strong>: A tool for infering scRNA_seq trajectory from RNA velocity</p>
<p><strong>Installation</strong>: The script works on python. R 3.6(or higher version) and “princurve” package in R are required. TF(Transfer entroy) inference needs openmpi installation on Linux OS. If you only want to infer trajectories of datasets, it is unnessary to install “openmpi” and “JPype”.</p>
<hr />
</div>
<div id="timeor" class="section level2">
<h2>6.2.2.7 <strong>TIMEOR</strong></h2>
<p><img src="/images/TIMEOR.png?width=25pc" /></p>
<p><strong>Introduction</strong>: Temporally unified RNA velocity for single cell trajectory inference (UniTVelo) is implementated on Python 3 and TensorFlow 2. The model estimates velocity of each gene and updates cell time based on phase portraits concurrently. The major features of UniTVelo are: 1)Using spliced RNA oriented design to model RNA velocity and transcription rates. 2) Introducing a unified latent time (Unified-time mode) across whole transcriptome to incorporate stably and monotonically changed genes. 3) Retaining gene-spcific time matrics (Independent mode) for complex datasets</p>
<p><strong>Installation</strong>: To run TIMEOR outside of website (recommended for preprocessing from raw .fastq files), users may use Docker and Docker Hub. First, the TIMEOR repository must be cloned (<a href="https://github.com/ashleymaeconard/TIMEOR.git" class="uri">https://github.com/ashleymaeconard/TIMEOR.git</a>). To use Docker, it must be installed (version 20.10.0 recommended).</p>
<p><strong>Web server</strong>: <a href="https://timeor.brown.edu/app/timeor">TIMEOR</a>.</p>
<hr />
</div>
<div id="unitvelo" class="section level2">
<h2>6.2.2.8 <strong>UniTVelo</strong></h2>
<p><img src="/images/UniTVelo.png?width=25pc" /></p>
<p><strong>Introduction</strong>: Temporally unified RNA velocity for single cell trajectory inference (UniTVelo) is implementated on Python 3 and TensorFlow 2. The model estimates velocity of each gene and updates cell time based on phase portraits concurrently. The major features of UniTVelo are: 1)Using spliced RNA oriented design to model RNA velocity and transcription rates. 2) Introducing a unified latent time (Unified-time mode) across whole transcriptome to incorporate stably and monotonically changed genes. 3) Retaining gene-spcific time matrics (Independent mode) for complex datasets</p>
<p><strong>Installation</strong>: The installation process is provided in <a href="https://github.com/StatBiomed/UniTVelo" class="uri">https://github.com/StatBiomed/UniTVelo</a></p>
<hr />
</div>
<div id="tips" class="section level2">
<h2>6.2.2.9 <strong>TIPS</strong></h2>
<p><strong>Introduction</strong>: Trajectory Inference of Pathway Significance through Pseudotime Comparison for Functional Assessment of single-cell RNAseq Data.</p>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code>install.packages(c(&quot;shiny&quot;,&quot;shinydashboard&quot;,&quot;markdown&quot;,&quot;ggplot2&quot;,&quot;Seurat&quot;,&quot;kohonen&quot;,&quot;viridis&quot;))

if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(c(&quot;monocle&quot;,&quot;switchde&quot;))

library(shiny)
shiny::runGitHub(&quot;TIPS&quot;, &quot;qingshanni&quot;) 

shiny::runApp(&quot;TIPS&quot;)</code></pre>
<hr />
</div>
<div id="velo-predictor" class="section level2">
<h2>6.2.2.10 <strong>Velo-Predictor</strong></h2>
<p><strong>Introduction</strong>: Velo-Predictor, an ensemble learning pipeline for RNA velocity prediction. While RNA velocity estimation is not straightforward, the pipeline can simplify the procedure by learning a predictive model from gene expression data.</p>
<p><strong>Installation</strong>: The installation process is provided in <a href="https://github.com/clay001/Velo-Predictor" class="uri">https://github.com/clay001/Velo-Predictor</a></p>
<hr />
</div>
<div id="origins" class="section level2">
<h2>6.2.2.11 <strong>ORIGINS</strong></h2>
<p><img src="/images/ORIGINS.png?width=15pc" /></p>
<p><strong>Introduction</strong>: Temporally unified RNA velocity for single cell trajectory inference (UniTVelo) is implementated on Python 3 and TensorFlow 2. The model estimates velocity of each gene and updates cell time based on phase portraits concurrently. The major features of UniTVelo are: 1)Using spliced RNA oriented design to model RNA velocity and transcription rates. 2) Introducing a unified latent time (Unified-time mode) across whole transcriptome to incorporate stably and monotonically changed genes. 3) Retaining gene-spcific time matrics (Independent mode) for complex datasets</p>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code>install.packages(&quot;remotes&quot;)

remotes::install_github(&quot;danielasenraoka/ORIGINS&quot;)</code></pre>
<hr />
</div>
<div id="paga" class="section level2">
<h2>6.2.2.12 <strong>PAGA</strong></h2>
<p><strong>Introduction</strong>: PAGA maps preserve the global topology of data, allow analyzing data at different resolutions, and result in much higher computational efficiency of the typical exploratory data analysis workflow. PAGA-initialized manifold learning algorithms converge faster, produce embeddings that are more faithful to the global topology of high-dimensional data, and introduce an entropy-based measure for quantifying such faithfulness. According to the <a href="https://www.nature.com/articles/s41587-019-0071-9">evaluation research</a> of trajectory methods, PAGA can model all trajectory types and performs better than any other method.</p>
<p><strong>Installation</strong>: PAGA is implemented in Scanpy via <a href="https://scanpy.readthedocs.io/en/latest/generated/scanpy.tl.paga.html">scanpy.tl.paga</a>. To use this method, please install <a href="https://scanpy.readthedocs.io/en/latest/index.html">Scanpy</a> first:</p>
<pre class="python"><code>pip install scanpy</code></pre>
<hr />
</div>
<div id="scorpius" class="section level2">
<h2>6.2.2.13 <strong>SCORPIUS</strong></h2>
<p><strong>Introduction</strong>: SCORPIUS, an unsupervised approach for inferring linear developmental chronologies from single-cell RNA sequencing data. In comparison to similar approaches, it has three main advantages:</p>
<ol style="list-style-type: decimal">
<li><p>It accurately reconstructs <strong>linear dynamic processes</strong>.</p></li>
<li><p>It automatically identifies marker genes, speeding up knowledge discovery.</p></li>
<li><p>It is fully unsupervised. Prior knowledge of the relevant marker genes or cellular states of individual cells is not required.</p></li>
</ol>
<p><strong>Installation</strong>: SCORPIUS can be used by installing the R package:</p>
<pre class="r"><code>install.packages(&quot;SCORPIUS&quot;)</code></pre>
<p>Or install the development version via:</p>
<pre class="r"><code>devtools::install_github(&quot;rcannood/SCORPIUS&quot;, build_vignettes = TRUE)</code></pre>
<hr />
</div>
<div id="stemnet" class="section level2">
<h2>6.2.2.14 <strong>STEMNET</strong></h2>
<p><strong>Introduction</strong>: STEMNET is used to estimate the progression of single stem cells to distinct lineages. As an input, it takes a single cell gene expression matrix and a vector annotating, for each cell, if the cell is part of a given developmental endpoint or if it an immature cell. It then estimates the direction and amount of priming for each stem cell and creates visualizations of the lineage priming process.</p>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
devtools::install_git(&#39;https://git.embl.de/velten/STEMNET/&#39;, build_vignettes=TRUE)</code></pre>
<hr />
</div>
<div id="monocle3" class="section level2">
<h2>6.2.2.15 <strong>Monocle3</strong></h2>
<p><strong>Introduction</strong>: The Monocle 3 package provides a toolkit for analyzing single-cell gene expression experiments, particularly as a powerful tool to infer cellular differentiation trajectory and dissect cell decisions during branch points. It is widely used by computational biologists and other researchers as it offers a streamlined analytical workflow for single-cell transcriptomics data, including three main types of analysis:</p>
<ol style="list-style-type: decimal">
<li>Clustering, classifying, and counting cells.</li>
<li>Constructing single-cell trajectories.</li>
<li>Differential expression analysis.</li>
</ol>
<p><strong>Installation</strong>: To install this package, start R and enter:</p>
<pre class="r"><code>### Install BiocManager first
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
install.packages(&quot;BiocManager&quot;)
BiocManager::install(version = &quot;3.14&quot;)

### Install dependencies
BiocManager::install(c(&#39;BiocGenerics&#39;, &#39;DelayedArray&#39;, &#39;DelayedMatrixStats&#39;,
                       &#39;limma&#39;, &#39;lme4&#39;, &#39;S4Vectors&#39;, &#39;SingleCellExperiment&#39;,
                       &#39;SummarizedExperiment&#39;, &#39;batchelor&#39;, &#39;HDF5Array&#39;,
                       &#39;terra&#39;, &#39;ggrastr&#39;))

### Install Monocle3
install.packages(&quot;devtools&quot;)
devtools::install_github(&#39;cole-trapnell-lab/monocle3&#39;)

### Install development version of monocle3
devtools::install_github(&#39;cole-trapnell-lab/monocle3&#39;, ref=&quot;develop&quot;)</code></pre>
<p><strong>Web server</strong>: <a href="https://cole-trapnell-lab.github.io/monocle3/">Monocle3</a>.
<img src="/images/Monocle3.jpg?width=60pc" alt="Monocle3" /></p>
<hr />
</div>
<div id="scvelo" class="section level2">
<h2>6.2.2.16 <strong>scVelo</strong></h2>
<p><strong>Introduction</strong>: scVelo employs a likelihood-based dynamical model to solve the full transcriptional dynamics of splicing kinetics, which distinguishes it with other algorithms obtaining velocities as the deviation of the observed ratio of spliced and unspliced mRNA from an inferred steady state. It also provides three existing approaches to perform RNA velocity estimation:</p>
<ol style="list-style-type: decimal">
<li>steady-state/deterministic model (using steady-state residuals)</li>
<li>stochastic model (using second-order moments),</li>
<li>dynamical model (using a likelihood-based framework).</li>
</ol>
<p><strong>Installation</strong>: To install this tool, start command line and enter:</p>
<pre class="bash"><code>### Install scVelo from PyPI using:
pip install -U scvelo

### Install development version of scVelo
pip install git+https://github.com/theislab/scvelo@main
### OR
git clone https://github.com/theislab/scvelo &amp;&amp; cd scvelo
git checkout --track origin/main
pip install -e .</code></pre>
<p><strong>Web server</strong>: <a href="https://scvelo.readthedocs.io/en/stable/">scvelo</a>.
<img src="/images/scVelo.jpg?width=60pc" alt="scvelo" /></p>
<hr />
</div>
<div id="velovi" class="section level2">
<h2>6.2.2.17 <strong>veloVI</strong></h2>
<p><strong>Introduction</strong>: veloVI (velocity variational inference) learns a gene-specifc dynamical model of RNA metabolism and provides a transcriptome-wide quantifcation of velocity uncertainty, which can be used for assessing whether RNA velocity analysis is suitable for a given dataset. It is also a flexible framework for modeling RNA dynamics by adapting the underlying dynamical model to use time-dependent transcription rates.</p>
<p><strong>Installation</strong>: To install this package, start command line and enter:</p>
<pre class="bash"><code>### Install the latest release on PyPI:
pip install velovi

### Install the latest development version:
pip install git+https://github.com/yoseflab/velovi.git@main</code></pre>
<p>It has been wrapped by <a href="https://docs.scvi-tools.org/en/stable/index.html">scvi-tools</a>, where you can find the API reference of veloVI.</p>
<p><strong>Web server</strong>: <a href="https://docs.scvi-tools.org/en/stable/api/reference/scvi.external.VELOVI.html#scvi.external.VELOVI">veloVI</a>.</p>
</div>
