---
title: "Enrichment analysis"
output:
  blogdown::html_page:
    toc: true
weight: 4
---

Enrichment analysis helps researchers gain mechanistic insight into gene lists generated from genome-scale (omics) experiments. This method identifies biological processes and pathways that are enriched in a gene list more than would be expected by chance.

**Breaking**: A plethora of enrichment tools linking different biological process and pathway databases are summarized.

---

## 1.4.1 **DAVID**

**Introduction**: DAVID is a popular bioinformatics resource system including a web server and web service for functional annotation and enrichment analyses of gene lists. It
consists of a comprehensive knowledgebase and a set of functional analysis tools.

![DAVID](/images/DAVID1.jpg?width=40pc)
 

**Installation**: DAVID web service allows users to programmatically (Java, Perl, Python, R) interact with the DAVID to automate user tasks. It is an enhancement and extension of current url-based DAVID API (https://david.ncifcrf.gov/content.jsp?file=WS.html) with more control and less limitations for users.

Download client code: 
[JavaClient-1.1.zip](https://david.ncifcrf.gov/webservice/sample_clients/JavaClient-1.1.zip) [(README.txt)](https://david.ncifcrf.gov/helps/WS_Java_README.txt), [PerlClient-1.1.zip](https://david.ncifcrf.gov/webservice/sample_clients/PerlClient-1.1.zip) [(README.txt)](https://david.ncifcrf.gov/helps/WS_perl_README.txt), [PythonClient-1.1.zip](https://david.ncifcrf.gov/webservice/sample_clients/PythonClient-1.1.zip) [(README.txt)](https://david.ncifcrf.gov/helps/WS_Python_README.txt), [Matlab-1.1.zip](https://david.ncifcrf.gov/webservice/sample_clients/Matlab-1.1.zip) [(note)](http://www.mathworks.com/support/solutions/en/data/1-AH2BDN/index.html?product=ML&solution=1-AH2BDN), [R package](http://bioinformatics.oxfordjournals.org/content/29/21/2810.abstract) (Please change the webaddress in the package to current DAVID address. See [Example in DAVID Forum](https://david-bioinformatics.freeforums.net/thread/63/compgo-david-registration-unsuccessful?page=1&scrollTo=166))
 
**Web server**: [DAVID](https://david.ncifcrf.gov/home.jsp).
![DAVID](/images/DAVID2.jpg?width=50pc)

---

## 1.4.2 **GSEA**

**Introduction**: Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states (e.g. phenotypes). The Molecular Signatures Database (MSigDB) is a collection of annotated gene sets for use with GSEA software.The Molecular(MSigDB) is a collection of annotated gene sets for use with GSEA software. The 33196 gene sets in the Human Molecular Signatures Database (MSigDB) are divided into 9 major collections, and several sub-collections. The 15918 gene sets in the Mouse Molecular Signatures Database (MSigDB) are divided into 6 major collections, and several sub-collections.
![GSEA1](/images/GSEA1.jpg?width=50pc)

**Installation**: R-GSEA makes GSEA available from the R programming environment, R implementation of GSEA that can be downloaded from the [Archived Downloads page](https://www.gsea-msigdb.org/gsea/downloads_archive.jsp) or [github](https://github.com/GSEA-MSigDB/GSEA_R).
```{r, eval=FALSE}
library("devtools")
install_github("GSEA-MSigDB/GSEA_R")

source(system.file('extdata', 'Run.GSEA.R', package = 'GSEA'))
```
GSEA Desktop is 100% Pure Java. Java 8 is required for our pre-built binaries. Builds against other versions of Java may be possible but are unsupported. Oracle Java is recommended as there are known issues when running with OpenJDK. See the [Downloads page](http://www.gsea-msigdb.org/gsea/downloads.jsp).

**Application**: A user guide was offered in [documentation page](https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html).

**Web server**: [GSEA](https://www.gsea-msigdb.org).
![GSEA2](/images/GSEA2.jpg?width=50pc)
 
---

## 1.4.3 **clusterProfiler**

**Introduction**: This package supports functional characteristics of both coding and non-coding genomics data for thousands of species with up-to-date gene annotation. It provides a univeral interface for gene functional annotation from a variety of sources and thus can be applied in diverse scenarios. It provides a tidy interface to access, manipulate, and visualize enrichment results to help users achieve efficient data interpretation. Datasets obtained from multiple treatments and time points can be analyzed and compared in a single run, easily revealing functional consensus and differences among distinct conditions.
 
**Installation**: To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("clusterProfiler")
```

**Application**: A full vignette was offered in [documentation page](https://yulab-smu.top/biomedical-knowledge-mining-book/index.html).

---


## 1.4.4 **Enrichr**

**Introduction**:Integrative web-based and mobile gene-list enrichment analysis tool that includes over 30 gene-set libraries, an alternative approach to rank enriched terms, and various interactive visualization approaches to display enrichment results using the JavaScript library Data Driven Documents (D3).
![Enrichr](/images/enrichr.jpg?width=50pc)

**Installation**: 
```{r, eval=FALSE}
library(devtools)
install_github("wjawaid/enrichR")
```
The package can be downloaded from CRAN using:
```{r, eval=FALSE}
install.packages("enrichR")
```

**Application**: A full vignette was offered in (https://cran.r-project.org/web/packages/enrichR/vignettes/enrichR.html).

**Web server**: [Enrichr](https://maayanlab.cloud/Enrichr/).

---

## 1.4.5 **KOBAS**

**Introduction**:KOBAS (KEGG Orthology Based Annotation System) is a web server for gene/protein functional annotation (Annotation module) and functional set enrichment (Enrichment module). Given a set of genes or proteins, it can determine whether a pathway, disease, or Gene Ontology(GO) term shows statistical significance. The current version is KOBAS 3.0 (also named KOBAS-i), which covers 5,944 species with incorporated knowledge.

KOBAS is composed of two functions, Annotation and Enrichment:gene list Enrichment,exp-data Enrichment.

**Installation**: Locally run using KOBAS docker version: Step 1: Download the KOBAS docker image; Step 2: Download the KOBAS sqlite3 DB and piptide sequences (ftp://ftp.cbi.pku.edu.cn/pub/KOBAS_3.0_DOWNLOAD/); Step 3: Enter into the KOBAS docker image. Run the command:
```{bash, eval=FALSE}
-v map path The path of two databases seq_pep and sqlite3, must be mapped to a path in the docker which will be used in the following parameters -q KOBASDB(--kobasdb=KOBASDB) and -y BLASTDB,(--blastdb=BLASTDB)
```

**Application**:  A user guide was offered in [help page](http://kobas.cbi.pku.edu.cn/help/).

**Web server**: [KOBAS](http://kobas.cbi.pku.edu.cn/).
 
---

## 1.4.6 **ToppGene**

**Introduction**: A one-stop portal for gene list enrichment analysis and candidate gene prioritization based on functional annotations and protein interactions network.
![ToppGene](/images/ToppGene.jpg?width=50pc)

**Application**:  A user guide was offered in [help page](https://toppgene.cchmc.org/help/help.jsp).

**Web server**: [ToppGene](https://toppgene.cchmc.org/).

---

## 1.4.7 **Metascape**

**Introduction**:  Metascape is a web-based portal designed to provide a comprehensive gene list annotation and analysis resource for experimental biologists. In terms of design features, Metascape combines functional enrichment, interactome analysis, gene annotation, and membership search to leverage over 40 independent knowledgebases within one integrated portal.
![Metascape](/images/Metascape.jpg?width=50pc)

**Application**:  A user guide was offered in [help page](https://metascape.org/gp/index.html#/menu/quickguide).

**Web server**: [Metascape](https://metascape.org/gp/index.html#/main/step1).

---

## 1.4.8 **WebGestalt**

**Introduction**: WebGestalt (WEB-based Gene SeT AnaLysis Toolkit) is a functional enrichment analysis web tool. Data sources for WebGestalt supports 12 organisms, 354 gene identifiers from various databases and technology platforms, and 321,251 functional categories from public databases and computational analyses. Experimental data from organisms or with gene identifiers not covered by the WebGestalt database can also be analyzed in WebGestalt. WebGestalt supports three well-established and complementary methods for enrichment analysis, including Over-Representation Analysis (ORA), Gene Set Enrichment Analysis (GSEA), and Network Topology-based Analysis (NTA).

**Installation**:
The package can be downloaded from CRAN 
[github](https://github.com/bzhanglab/WebGestaltR) using:
```{r, eval=FALSE}
install.packages("WebGestaltR")
```

**Application**:  A user guide was offered in [Manual page](http://www.webgestalt.org/WebGestalt_2019_Manual.pdf).

**Web server**: [WebGestalt](http://www.webgestalt.org/).
![WebGestalt](/images/WebGestalt.jpg?width=50pc)

---

## 1.4.9 **STEM**

**Introduction**: The Short Time-series Expression Miner (STEM) is a Java program for clustering, comparing, and visualizing short time series gene expression data from microarray experiments (~8 time points or fewer). STEM allows researchers to identify significant temporal expression profiles and the genes associated with these profiles and to compare the behavior of these genes across multiple conditions. STEM is fully integrated with the Gene Ontology (GO) database supporting GO category gene enrichment analyses for sets of genes having the same temporal expression pattern. STEM also supports the ability to easily determine and visualize the behavior of genes belonging to a given GO category or user defined gene set, identifying which temporal expression profiles were enriched for these genes. 

**Installation**: STEM version 1.3.13 (version log) is freely available under a GPL v3.0 license. Click [here](https://www.cs.cmu.edu/~jernst/stem/stem.zip) to download STEM.Source code is available on [gitHub](http://sb.cs.cmu.edu/stem/screenshots.html).

**Application**:  A short video showing screenshots of STEM can be found [here](http://sb.cs.cmu.edu/stem/stem.avi).

**Web server**: [WebGestalt](http://sb.cs.cmu.edu/stem/).

---

## 1.4.10 **GAGE**

**Introduction**: GAGE is a published method for gene set (enrichment or GSEA) or pathway analysis. GAGE is generally applicable independent of microarray or RNA-Seq data attributes including sample sizes, experimental designs, assay platforms, and other types of heterogeneity, and consistently achieves superior performance over other frequently used methods. In gage package, we provide functions for basic GAGE analysis, result processing and presentation. We have also built pipeline routines for of multiple GAGE analyses in a batch, comparison between parallel analyses, and combined analysis of heterogeneous data from different sources/studies. In addition, gage provide demo microarray data and commonly used gene set data based on KEGG pathways and GO terms. These funtions and data are also useful for gene set analysis using other methods.

**Installation**:
To install this package, start R (version "4.2") and enter:
```{r eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("gage")
```

**Application**:  A tutorial was offered in [here](https://bioconductor.org/packages/release/bioc/vignettes/gage/inst/doc/RNA-seqWorkflow.pdf).

---

## 1.4.11 **GeneCodis**

**Introduction**: GeneCodis is an API and web tool for the Singular and Modular Enrichment Analysis (SEA and MEA) of lists of genes, proteins, and regulatory elements: miRNAs, transcription factors (TFs), and CpGs. Look CoAnnotation in the next panel for more information on the Modular Enrichment Analysis. The enrichment analysis of regulatory elements, requires a different approach. Due to a gene selection bias (some genes are more targeted than others) in platforms and regulation networks it requires a test based on a non-central hypergeometric distribution, concretely, we use the Wallenius distribution. Our method is based on the GOseq implementation. Currently GeneCodis provides annotations for 14 different organisms and gather 21 annotation sources. These are updated every three months. 

**Installation**:The GeneCodis API is a set of endpoints focused for developers who want to programmatically use the GeneCodis features. The documentation of the GeneCodis API is published by Postman, an API platform that helps users and simplifies processes like design or testing. This documentation is available in C#, cURL, Dart, GO, Java, Javascript, C, NodeJS, Objective-C, Ocaml, PHP, Powershell, Python, R, Ruby, Shell and Swift. See the details in the [GeneCodis API documentation](https://documenter.getpostman.com/view/21704552/2s7YfPfuPN) and [github](https://github.com/GENyO-BioInformatics/GeneCodis).

**Application**:  A user guide was offered in [help page](https://genecodis.genyo.es/).

**Web server**: [GeneCodis](https://genecodis.genyo.es/).
![GeneCodis](/images/GeneCodis.jpg?width=50pc)

---

## 1.4.12 **GeneTrail**

**Introduction**: GeneTrail is a web-interface providing access to different tools for the statistical analysis of molecular signatures. GeneTrail allows the integrated analysis of transcriptomic, miRNomic, genomic, and proteomic datasets. It offers multiple statistical tests, a large number of predefined reference sets, as well as a comprehensive collection of biological categories and enables direct comparisons between the computed results.

**Installation**: Currently this tool only support building GeneTrail 3 under Linux in [github](https://github.com/unisb-bioinf/genetrail3). Windows and MacOS X are not supported.

**Application**:  A user guide was offered in [Documentation page](https://genetrail.bioinf.uni-sb.de/help?topic=parameter_information).

**Web server**: [GeneTrail](https://genetrail.bioinf.uni-sb.de/start.html).
![GeneTrail](/images/GeneTrail.jpg?width=50pc)

---

## 1.4.13 **PageMan**

**Introduction**: PageMan is a tool to get a quick overview of multiparallel experiments. PageMan also helps comparing experiments from different organisms.

**Installation**: This tool offered three install version:[Linux Installer](https://mapman.gabipd.org/pageman?p_p_id=MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6&p_p_lifecycle=0&p_p_state=maximized&p_p_mode=view&p_p_col_id=column-3&p_p_col_count=2&_MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6_ItemId=28&_MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6_Download=Item), [MacOSX Installer](https://mapman.gabipd.org/pageman?p_p_id=MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6&p_p_lifecycle=0&p_p_state=maximized&p_p_mode=view&p_p_col_id=column-3&p_p_col_count=2&_MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6_ItemId=29&_MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6_Download=Item), [Windows Installer](https://mapman.gabipd.org/pageman?p_p_id=MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6&p_p_lifecycle=0&p_p_state=maximized&p_p_mode=view&p_p_col_id=column-3&p_p_col_count=2&_MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6_ItemId=30&_MapManDownload_WAR_MapManDownloadportlet_INSTANCE_wtL6_Download=Item).

**Web server**: [PageMan](https://mapman.gabipd.org/pageman).

---

## 1.4.14 **ErmineJ**

**Introduction**: ErmineJ performs analyses of gene sets in high-throughput genomics data such as gene expression profiling studies. A typical goal is to determine whether particular biological pathways are "doing something interesting" in an experiment that generates long lists of candidates. The software is designed to be used by biologists with little or no informatics background.

**Installation**: ErmineJ is stand-alone (desktop) software. It does not run in a web browser. All data and analysis results remain on your computer and are not uploaded anywhere. If you want to install ErmineJ “locally”, see the "SOFTWARE FILES": 

  | Description     | File           | 
  | :-------------| :--------------| 
  | Windows installer  | [ermineJ-3.2-setup.exe](http://home.pavlab.msl.ubc.ca/ermineJ/distributions/ermineJ-3.2-setup.exe) |
  | Mac OSX application package  | [ermineJ-3.2.dmg](http://home.pavlab.msl.ubc.ca/ermineJ/distributions/ermineJ-3.2.dmg) |
  | Generic bundle | [ermineJ-3.2-generic-bundle.tar.gz](http://home.pavlab.msl.ubc.ca/ermineJ/distributions/ermineJ-3.2-generic-bundle.zip) |
  | Source code | [ermineJ-3.2-src.tar.gz](http://home.pavlab.msl.ubc.ca/ermineJ/distributions/ermineJ-3.2-src.tar.gz)|
  
ermineR requries 64 bit version of java to function. If you are a Mac user make sure you have the java SDK. After java is installed you can install ermineR by doing.  the details in the [github](https://github.com/PavlidisLab/ermineR).
```{r, eval=FALSE}
devtools::install_github('PavlidisLab/ermineR')
```

**Application**:  A user guide was offered in [Documentation page](https://erminej.msl.ubc.ca/help/tutorials/).

**Web server**: [ErmineJ](https://erminej.msl.ubc.ca/).
![ErmineJ](/images/ErmineJ.jpg?width=50pc)

---

## 1.4.15 **EnrichNet**

**Introduction**: EnrichNet is a web-application that complements this classical overlap-based enrichment analysis by a new association measure that integrates information from the known network structure of interactions between proteins. 

**Installation**: The Python script below submits a gene list consisting of 5 genes in ENSEMBL-format to EnrichNet, for an enrichment analysis on the BioCarta database and using molecular interaction data from the STRING database.
```{python, eval = FALSE}

#!/usr/bin/env python

# The urllib and urllib2 libraries need to be installed!
import urllib
import urllib2

# Here, the genes, the identifier format, the pathway database and the network are specified
genes = 'ENSG00000113916,ENSG00000068024,ENSG00000108840,ENSG00000061273,ENSG00000005339'
idtype = 'ensembl'
pathdb = 'biocarta'
graph = 'string'

url = 'http://www.enrichnet.org/index.php'
params = urllib.urlencode({  'genestr':genes,  'idtype':idtype, 'pathdb':pathdb, 'graph':graph})

# As a result a hyperlink to a temporary web-page is generated, on which the user will be redirected
# to the interactive results page, once the results are available
response = urllib2.urlopen(url+'?'+params).read()
```
Python example source code download [here](http://www.enrichnet.org/enrichnet_upload_script.zip).

**Application**:  A user guide was offered in [tutorial page](http://www.enrichnet.org/tutorial.html?SESSION_NAME=SESS63ccebf95f5e1).

**Web server**: [EnrichNet](http://www.enrichnet.org/).
![EnrichNet](/images/EnrichNet.jpg?width=50pc)

---

## 1.4.16 **DOSE**

**Introduction**: This package implements five methods proposed by Resnik, Schlicker, Jiang, Lin and Wang respectively for measuring semantic similarities among DO terms and gene products. Enrichment analyses including hypergeometric model and gene set enrichment analysis are also implemented for discovering disease associations of high-throughput biological data.

**Installation**: DOSE is released within the Bioconductor project and the source code is hosted in [GitHub](https://github.com/GuangchuangYu/DOSE).
```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DOSE")

browseVignettes("DOSE")
```

**Application**:  A vignettes was offered in [here](https://guangchuangyu.github.io/software/DOSE/documentation/).

**Web server**: [DOSE](https://guangchuangyu.github.io/software/DOSE/).

---

## 1.4.17 **SciMiner**

**Introduction**: SciMiner is a web-based literature mining and functional analysis tool that identifies genes and proteins using a context specific analysis of MEDLINE abstracts and full texts. SciMiner accepts a free text query string (PubMed Entrez search) or a list of PubMed Unique Identifiers (PMIDs) as input.

**Installation**: Standalone SciMiner is available in two different formats. Read the installation guide first and determine which package would be more appropriate for you.

 | Description     | File           | 
  | :-------------| :--------------| 
  | Package-I  | [Linux with MySQL, webserver](javascript:download_package_I();) |
  | Package-II | [VMWare workstation or VMplayer](javascript:download_package_II();) |

**Application**:  A user manual was offered in [here](http://hurlab.med.und.edu/SciMiner/) and a installation guide was offered in [here](http://hurlab.med.und.edu/SciMiner/Files/SciMiner_Installation_Guide_v3.1_2010-01-15.pdf).

**Web server**: [SciMiner](http://hurlab.med.und.edu/SciMiner/).

---

## 1.4.18 **ProfCom**

**Introduction**: ProfCom is a web-based tool for the interpretation of genes that were identified to be functionally linked by experiment. ProfCom automatically generates statistically valid hypotheses / biological models from gene or protein lists supplied by the user. This knowledge can help to improve your understanding of genetic mechanisms involved in biological phenomena under study.

**Application**:  A user manual was offered in [here](http://webclu.bio.wzw.tum.de/profcom/demo.php).

**Web server**: [ProfCom](http://webclu.bio.wzw.tum.de/profcom/).

---

## 1.4.19 **hypeR**

**Introduction**: The hypeR package goes beyond performing basic enrichment, by providing a suite of methods designed to make routine geneset enrichment seamless for scientists working in R.

**Installation**: To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("hypeR")
```
Install the development version of the package from [Github](https://montilab.github.io/hypeR-docs/).
```{r, eval=FALSE}
devtools::install_github("montilab/hypeR")
```
Or install with Conda.
```{bash, eval=FALSE}
conda create --name hyper
source activate hyper
conda install -c r r-devtools
R
library(devtools)
devtools::install_github("montilab/hypeR")
```

**Application**:  A user manual was offered [here](https://www.bioconductor.org/packages/release/bioc/vignettes/hypeR/inst/doc/hypeR.html).

---

## 1.4.20 **CIE**

**Introduction**: Causal Inference Enrichment (CIE) is a platform for identification of active transcriptional regulators from differential gene expression data. CIE provides directional (Ternary and Quaternary scoring statistic) and un-directional (Fisher's Exact) enrichment analysis on regulatory networks. Networks integrated within the platform include TRED, TRRUST, STRINGdb, ChIP-Atlas, and tissue corrected ChIP-Atlas. TRED and ChIP-Atlas are not annotated with information on the mode of regulation (activation vs. repression) and are suitable for use with Fisher's exact test only. Tissue corrected ChIP-Atlas is a network constructed by combining ChIP-Atlas with tissue-specific gene expression data from GTEx using a regularized Gaussian Graphical Model. This network is annotated with information on the mode of regulation and can be used with Fisher or Ternary enrichment methods. STRINGdb has mixed directionality and can be used with the Quaternary method.

**Installation**: You can easily run CIE on your own machine. The data available for download to your left is the human subset of the database named in the title. For information on our installing our R package, and pulling from databases with different parameters (ex. species) to suit your experiment, please visit [GitHub](https://github.com/umbibio/CIE-R-Package).
```{r, eval=FALSE}
install.packages("BiocManager")

BiocManager::install("org.Hs.eg.db")
```

**Application**:  A user manual was offered in [help page](https://umbibio.math.umb.edu/cie/app).

**Web server**: [CIE](https://umbibio.math.umb.edu/cie/app).
![CIE](/images/CIE.jpg?width=50pc)

---

## 1.4.21 **GeneAnswers**

**Introduction**: GeneAnswers provides an integrated tool for biological or medical interpretation of the given one or more groups of genes by means of statistical test.

**Installation**: To install this package, start R and enter:
```{r, eval=FALSE}
## try http:// if https:// URLs are not supported
source("https://bioconductor.org/biocLite.R")
biocLite("GeneAnswers")
```

**Application**:  A user manual was offered in [here](https://www.bioconductor.org/packages//2.11/bioc/manuals/GeneAnswers/man/GeneAnswers.pdf).

---

## 1.4.22 **SIGNAL**

**Introduction**: The SIGNAL platform is designed to facilitate robust hit selection from normalized high-throughput datasets.

**Installation**: Download SIGNAL as a standalone [R function](https://signal.niaid.nih.gov/RscriptDownload/SIGNAL_R_function.R).

**Application**:  A user manual was offered in [User Guide](https://signal.niaid.nih.gov/).

**Web server**: [SIGNAL](https://signal.niaid.nih.gov/).
![SIGNAL](/images/SIGNAL.jpg?width=50pc)

---

## 1.4.23 **Network2Canvas**

**Introduction**: Network2Canvas (N2C), a web application that provides an alternative way to view networks. N2C visualizes network nodes by placing them on a square toroidal grid. These nodes are then clustered together on the grid using simulated annealing to maximize local connections. For visualizing the canvas, a node’s brightness is made proportional to its local fitness; the brighter a node is, the stronger its connections to its neighbors. The grid is interactive, implemented in HTML5 and the Java Script library D3. 

**Installation**: This [zipped Python package ](http://www.maayanlab.net/N2C/res/Network2Canvas%20Script%20Package.zip) contains the Network2Canvas script and other supporting scripts. The package also includes example input files.

**Application**:  A user manual was offered in [Help page](http://www.maayanlab.net/N2C/help.html).

**Web server**: [Network2Canvas](http://www.maayanlab.net/N2C).
![Network2Canvas](/images/Network2Canvas.jpg?width=50pc)

---

## 1.4.24 **HPOSim**

**Introduction**: This package implements multiple similarity measures for HPO terms, genes and diseases. It is aiming at phenotype analysis for gene sets and disease sets. Functions for HPO enrichment analysis is also provided.

**Installation**: To install this package, start R and enter:
```{r, eval=FALSE}
install.packages("HPOSim")
```

**Application**:  A user manual was offered in [here](http://www2.uaem.mx/r-mirror/web/packages/HPOSim/HPOSim.pdf).

---

# GO Enrichment##

---

## 1.4.25 **BiNGO**

**Introduction**: BiNGO is a Java-based tool to determine which Gene Ontology (GO) categories are statistically overrepresented in a set of genes or a subgraph of a biological network. BiNGO is implemented as a plugin for Cytoscape, which is a an open source bioinformatics software platform for visualizing and integrating molecular interaction networks. BiNGO maps the predominant functional themes of a given gene set on the GO hierarchy, and outputs this mapping as a Cytoscape graph. Gene sets can either be selected or computed from a Cytoscape network (as subgraphs) or compiled from sources other than Cytoscape (e.g. a list of genes that are significantly upregulated in a microarray experiment). The main advantage of BiNGO over other GO tools is the fact that it can be used directly and interactively on molecular interaction graphs. Another plus is that BiNGO takes full advantage of Cytoscape's versatile visualization environment. This allows you to produce customized high-quality figures.

**Installation**: To obtain BiNGO, you must install Cytoscape 3.0 or higher (www.cytoscape.org) and install BiNGO using the [Cytoscape App Manager](https://apps.cytoscape.org/apps/bingo). The source code of BiNGO is available at [github](https://github.com/idekerlab/BiNGO).

**Application**: The BiNGO tutorial was offered in [here](https://www.psb.ugent.be/cbd/papers/BiNGO/Tutorial.html).

**Web server**: [BiNGO](https://www.psb.ugent.be/cbd/papers/BiNGO/Home.html).

---

## 1.4.26 **GOrilla**

**Introduction**: GOrilla is a tool for identifying and visualizing enriched GO terms in ranked lists of genes.

**Application**: The tutorial was offered in [here](http://cbl-gorilla.cs.technion.ac.il/help.html).

**Web server**: [GOrilla](http://cbl-gorilla.cs.technion.ac.il/).

---

## 1.4.27 **WEGO**

**Introduction**: WEGO (Web Gene Ontology Annotation Plot) is a simple but useful tool for visualizing, comparing and plotting GO (Gene Ontology) annotation results. As the GO vocabulary became more and more popular, WEGO was widely adopted and used in many researches. 
![WEGO](/images/WEGO.jpg?width=50pc)

**Application**: The WEGO tutorial was offered in [here](https://wego.genomics.cn/document).

**Web server**: [WEGO](https://wego.genomics.cn/).

---

## 1.4.28 **GOstats**

**Introduction**: WEGO (Web Gene Ontology Annotation Plot) is a simple but useful tool for visualizing, comparing and plotting GO (Gene Ontology) annotation results. As the GO vocabulary became more and more popular, WEGO was widely adopted and used in many researches. 

**Installation**: To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("GOstats")
```

**Application**: A user manual was offered in [here](https://bioconductor.org/packages/release/bioc/vignettes/GOstats/inst/doc/GOvis.pdf).

---

## 1.4.29 **GO Term Finder**

**Introduction**: The GO Term Finder (Version 0.86) searches for significant shared GO terms, or parents of those GO terms, used to describe the genes in your list to help you discover what the genes may have in common. To map annotations of a group of genes to more general terms and/or to bin them in broad categories, use the GO Slim Mapper.

**Web server**: [GO Term Finder](https://www.yeastgenome.org/goTermFinder).
![GOTermFinder](/images/GOTermFinder.jpg?width=50pc)

---

## 1.4.30 **topGO**

**Introduction**: topGO package provides tools for testing GO terms while accounting for the topology of the GO graph. Different test statistics and different methods for eliminating local similarities and dependencies between GO terms can be implemented and applied.

**Installation**: To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("topGO")
```

**Application**: A user manual was offered [here](https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf).

---

## 1.4.31 **GOEAST**

**Introduction**: GOEAST is web based software toolkit providing easy to use, visualizable, comprehensive and unbiased Gene Ontology (GO) analysis for high-throughput experimental results, especially for results from microarray hybridization experiments. The main function of GOEAST is to identify significantly enriched GO terms among give lists of genes using accurate statistical methods.
![GOEAST1](/images/GOEAST1.jpg?width=50pc)

**Application**: The tutorial was offered in [here](http://omicslab.genetics.ac.cn/GOEAST/tutorial.php).

**Web server**: [GOEAST](http://omicslab.genetics.ac.cn/GOEAST/).
![GOEAST2](/images/GOEAST2.jpg?width=50pc)

---

## 1.4.32 **FuncAssociate**

**Introduction**: FuncAssociate is a web-based tool to help researchers use Gene Ontology attributes to characterize large sets of genes derived from experiment.

**Application**: The tutorial was offered in [here](http://llama.mshri.on.ca/funcassociate/documentation).

**Web server**: [FuncAssociate](http://llama.mshri.on.ca/funcassociate/).
![FuncAssociate](/images/FuncAssociate.jpg?width=50pc)

---

## 1.4.33 **Ontologizer**

**Introduction**: Ontologizer is a tool for the statistical analysis and visualization of high-throughput biological data using Gene Ontology.
 
**Installation**:  It is available as a Java program for both UI and command line usage as well as JavaScript version that can be conveniently run in modern Web browsers. See the details in the [github](https://github.com/sba1/ontologizer).

**Application**: The tutorial was offered in [here](http://llama.mshri.on.ca/funcassociate/documentation).

**Web server**: [Ontologizer](http://ontologizer.de/webontologizer/).
![Ontologizer](/images/Ontologizer.jpg?width=50pc)

---

## 1.4.34 **GOSim**

**Introduction**: This package implements several functions useful for computing similarities between GO terms and gene products based on their GO annotation. Moreover it allows for computing a GO enrichment analysis.

**Installation**:  To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("GOSim")
```

**Application**: The manual was offered in [here](https://www.bioconductor.org/packages/release/bioc/vignettes/GOSim/inst/doc/GOSim.pdf).

---

## 1.4.35 **GoSurfer**

**Introduction**: GoSurfer uses Gene Ontology (GO) information to analyze gene sets obtained from genome-wide computations or microarray analyses. GoSurfer is a graphical interactive data mining tool. It associates user input genes with GO terms and visualizes such GO terms as a hierarchical tree. Users can manipulate the tree output by various means, like setting heuristic thresholds or using statistical tests. Significantly important GO terms resulted from a statistical test can be highlighted. All related information are exportable either as texts or as graphics.

**Application**: The tutorial was offered in [here](https://systemsbio.ucsd.edu/GoSurfer/tutorial.htm).

**Web server**: [GoSurfer](https://systemsbio.ucsd.edu/GoSurfer/).

---

## 1.4.36 **ShinyGO**

**Introduction**: ShinyGO is an intuitive, graphical web application that can help researchers gain actionable insights from gene-sets based on a large annotation database derived from Ensembl and STRING-db for 59 plant, 256 animal, 115 archeal and 1678 bacterial species.
![ShinyGO](/images/ShinyGO.jpg?width=50pc)

**Web server**: [ShinyGO](http://bioinformatics.sdstate.edu/go/).

---

## 1.4.37 **ADGO**

**Introduction**: ADGO provides biological interpretations for microarray data and list of genes (or proteins) via composite annotations.

**Application**: The tutorial was offered in [here](http://www.btool.org/adgo2_dir/ADGO_2_how_to_use.docx).

**Web server**: [ADGO](http://www.btool.org/ADGO2).
![ADGO](/images/ADGO.jpg?width=50pc)

---

## 1.4.38 **tmod**

**Introduction**: The package includes several tools for testing the significance of enrichment of modules or other gene sets as well as visualisation of the features (genes, metabolites etc.) and gene sets, feature sets or transcriptional modules. 

**Installation**:  To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
install.packages("tmod")
```

**Application**: The vignettes was offered in [here](https://cran.r-project.org/web/packages/tmod/vignettes/tmod.html).

---

## 1.4.39 **modPhEA**

**Introduction**: model organism Phenotype Enrichment Analysis (modPhEA) is a freely available tool reports phenotypes that are associated with given gene sets. Currently, there are 132 animal and 50 fungal species which can be analyzed by modPhEA.

**Application**: The tutorial was offered in [here](https://evol.nhri.org.tw/phenome2/tutorial.jsp#index21).

**Web server**: [modPhEA](https://evol.nhri.org.tw/phenome2/).
![modPhEA](/images/modPhEA.jpg?width=50pc)

---

## 1.4.40 **netGO**

**Introduction**: netGO is an R/Shiny package for network-integrated pathway enrichment analysis.netGO provides user-interactive visualization of enrichment analysis results and related networks.

**Installation**: To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}

#The user may want to use the following codes to install the required packages.

install.packages('devtools') # 1.4.1
library(devtools) # check Rcpp package is installed.
install_github('unistbig/shinyCyJS')
install.packages('doParallel') # 1.0.15
install.packages('doSNOW') # 1.0.18
install.packages('DT') # 0.11
install.packages('foreach') # 1.4.7
install.packages('googleVis') # 0.6.4
install.packages('htmlwidgets') # 1.5.1
install.packages('shiny') # 1.4.0
install.packages('shinyjs') # 1.0
install.packages('V8') # 2.3

library(devtools)
install_github('unistbig/netGO') # install netGO library

library(netGO) # load netGO library
DownloadExampleData() # Download and load the breast tumor data
obj = netGO(genes = brca[1:30], genesets, network, genesetV) 

# The user may also load the pre-calculated result using the following command
# load("brcaresult.RData")   
```

**Application**: The vignettes was offered in [here](https://github.com/unistbig/netGO).

---

## 1.4.41 **FunSet**

**Introduction**: The FunSet webserver performs Gene Ontology (GO) enrichment analysis, identifying GO terms that are statistically overrepresented in a target set with respect to a background set. The enriched terms are displayed in a 2D plot that captures the semantic similarity between terms, with the option to cluster the terms and identify a representative term for each cluster. FunSet can be used interactively or programmatically, and allows users to download the enrichment results both in tabular form and in graphical form as SVG files or in data format as JSON or csv.

**Web server**: [FunSet](http://funset.uno/).
![FunSet](/images/modPhEA.jpg?width=50pc)
 
---

# Pathway Enrichment##

---

## 1.4.42 **SPEED2**

**Introduction**: SPEED2 is a signaling pathway annotation enrichment analysis tool with gene sets derived from manually curated pathway perturbation experiments in human cell lines. Identifying modulated pathways upstream of differentially expressed genes can facilitate the understanding of involved regulatory mechanisms. Currently, only human genes and pathways are supported. SPEED2 gene sets are extracted from publically available microarray data from the GEO database, which have been selected for pathway specific perturbations. This new version contains 16 instead of 11 pathways, and is based on three times more experimental data sets. It also uses different statistical methods to derive and score signatures.

**Application**: The SPEED2 tutorial was offered in [here](https://speed2.sys-bio.net/).

**Web server**: [SPEED2](https://speed.sys-bio.net/).
![SPEED2](/images/SPEED2.jpg?width=50pc)
 
---

## 1.4.43 **TACCO**

**Introduction**: TACCO is a database designed specifically for connecting transcriptome alterations, pathway alterations and clinical outcomes in cancers. Users can either select differentially expressed genes with different criteria within the TACCO database or upload their own gene list to be analyzed. Users can then perform KEGG pathway enrichment analysis, Gene ontology analysis or construct a multiple gene model for the prognosis prediction using this gene list.

**Application**: The tutorial was offered in [here](http://tacco.life.nctu.edu.tw/).

**Web server**: [TACCO](http://tacco.life.nctu.edu.tw/).

---

## 1.4.44 **PathVisio**

**Introduction**: PathVisio is a free open-source pathway analysis and drawing software which allows drawing, editing, and analyzing biological pathways. It is developed in Java and can be extended with plugins.

**Installation**: Installation PathVisio by [Installation Guide](https://pathvisio.org/tutorials/installation_guide.html);
Current version: [PathVisio 3.3.0](https://github.com/PathVisio/pathvisio/releases/download/v3.3.0/pathvisio_bin-3.3.0.zip).
**Application**: The tutorial was offered in [here](https://pathvisio.org/tutorials.html).

**Web server**: [PathVisio](http://tacco.life.nctu.edu.tw/).

---

## 1.4.45 **NetWalker**

**Introduction**: NetWalker is a desktop application for functional analyses of large-scale genomics datasets within the context of molecular networks.

**Installation**: Download Click [here](https://netwalkersuite.org/download) to find the NetWalker installer, NetWalker manual, and sample datasets for use with the application.

**Application**: The tutorial was offered in [here](https://netwalkersuite.org/tutorials/doxorubicin/data-import-and-processing).

**Web server**: [NetWalker](https://netwalkersuite.org/).

---

## 1.4.46 **Graphite Web**

**Introduction**: Graphite web is a public web server for the analysis and visualization of biological pathways using high-throughput gene expression data. It supports five different gene set analysis, three species and two databases of pathways. Graphite web has a powerful pathway visualization that makes results interpretation easily accessible to the user.


**Application**: The tutorial was offered in [here](https://graphiteweb.bio.unipd.it/tutorial.html).

**Web server**: [Graphite Web](https://graphiteweb.bio.unipd.it/index.html).
![GraphiteWeb](/images/GraphiteWeb.jpg?width=50pc)
 
---

## 1.4.47 **Paintomics**

**Introduction**: Paintomics is a web tool for the integrative visualization of multiple omic datasets onto KEGG (102.0 April 1 2022), Reactome (Version 76 21 March 2021) and MapMan (GoMapMan 25 May 2018) biological pathway maps. Currently, Paintomics supports integrated visualization of multiple species of different biological kingdoms and offers user the possibility to request any other organism present in the KEGG, Reactome and MapMan database. We will update all those databases annually.

**Installation**: In addition to using the public PaintOmics 4 servers, you can install your own instance of PaintOmics 4. Download the latest source code from GitHub:
```{bash, eval=FALSE}
cd
git clone https://github.com/fikipollo/paintomics3.git
ls $PWD/paintomics3
```

**Application**: The tutorial was offered in [here](https://paintomics.readthedocs.io/en/latest/).

**Web server**: [PaintOmics](https://www.paintomics.org/).
![PaintOmics](/images/PaintOmics.jpg?width=50pc)
 
---

## 1.4.48 **HisCoM-PAGE**

**Introduction**: Hierarchical Structural Component Models of Pathway A nalysis using Gene Expression data (HisCoM-PAGE) is a novel statistical approach for an analysis of gene expression data using pathway information.

**Installation**: HisCoM-PAGE is freely available for download.
All source code is written in R code: [Download zip](http://statgen.snu.ac.kr/software/HisCom-PAGE/Download.zip) and [Sample data zip](http://statgen.snu.ac.kr/software/HisCom-PAGE/Sample_input_data.zip).

**Application**: The tutorial was offered in [here](http://statgen.snu.ac.kr/software/HisCom-PAGE/tutorial.html).

**Web server**: [HisCoM-PAGE](http://statgen.snu.ac.kr/software/HisCom-PAGE/).
 
---

## 1.4.49 **PAGWAS**

**Introduction**: Bayesian hierarchical methods for pathway analysis of genomewide association data: Normal/Bayes factors and Sparse Normal/Adaptive lasso. The Frequentist Fisher’s method is included as well.

**Installation**: To install this package, start R (version "4.2") and enter:
```{r, eval=FALSE}
install.packages("PAGWAS")
```

**Application**: The tutorial was offered in [here](https://cran.r-project.org/web/packages/PAGWAS/PAGWAS.pdf).

---

## 1.4.50 **DecoPath**

**Introduction**: DecoPath is a scientific web application that has been developed in an academic capacity and thus comes with no warranty or guarantee of maintenance, support, or back-up of data.

**Installation**: Clone and download DecoPath from [GitHub](https://github.com/DecoPath/DecoPath):
```{bash, eval=FALSE}
$ git clone https://github.com/decopath/decopath.git
$ cd decopath

#Generate a SECRET_KEY for password encryption using the code below and add the generated text to the settings file on line 23.
$ python3.7 generate_secret_key.py

#If you are not using a domain name, remove the string domain.com from line 29 of the settings file and line 45 of the docker-compose file.

#Additionally, you can choose to update the admin email address in the #docker-compose file on line 14. Note the default admin password: admin.

#Build the DecoPath image using Docker.
$ docker image build -t decopath:latest .

#Start the docker image using docker-compose.
$ docker-compose up -d
```
 DecoPath is provided as a Python 3.7 package that you can customize and edit the functionality of. You can clone the repository from GitHub and set up the website following the commands below:
```{python, eval=FALSE}
$ git clone https://github.com/decopath/decopath.git
$ cd decopath
$ python3.7 -m pip install -r requirements.txt
$ python3.7 manage.py makemigrations
$ python3.7 manage.py migrate
$ python3.7 manage.py load_db
$ python3.7 manage.py init_user --superuser "user@domain.com"
$ python3.7 manage.py runserver
```

**Application**: The vignettes was offered in [here](https://github.com/DecoPath/DecoPath).

---
