---
title: "Clustering Analysis"
output:
  blogdown::html_page:
    toc: true
weight: 2
---


<div id="TOC">
<ul>
<li><a href="#nbclust" id="toc-nbclust">6.2.1 <strong>Nbclust</strong></a></li>
<li><a href="#clustvis" id="toc-clustvis">6.2.2 <strong>ClustVis</strong></a></li>
<li><a href="#medusa" id="toc-medusa">6.2.3 <strong>Medusa</strong></a></li>
<li><a href="#wcd" id="toc-wcd">6.2.4 <strong>wcd</strong></a></li>
<li><a href="#timeclust" id="toc-timeclust">6.2.5 <strong>TimeClust</strong></a></li>
<li><a href="#hiplot" id="toc-hiplot">6.2.6 <strong>hiplot</strong></a></li>
<li><a href="#cola" id="toc-cola">6.2.7 <strong>cola</strong></a></li>
</ul>
</div>

<p>Unsupervised or supervised clustering analysis has been an integral part of transcriptomics, playing an important role in the identification of sub-groups.</p>
<p><strong>Breaking</strong>: Some well-used and comprehensive clustering tools are provided in this section.</p>
<hr />
<div id="nbclust" class="section level2">
<h2>6.2.1 <strong>Nbclust</strong></h2>
<p><strong>Introduction</strong>: The R package provides 30 indices which determine the number of clusters in a data set and it offers also the best clustering scheme from different results to the user. In addition, it provides a function to perform k-means and hierarchical clustering with different distance measures and aggregation methods. Any combination of validation indices and clustering methods can be requested in a single function call. This enables the user to simultaneously evaluate several clustering schemes while varying the number of clusters, to help determining the most appropriate number of clusters for the data set of interest.</p>
<p><strong>Installation</strong>: To install this package, start R (version “4.2”) and enter:</p>
<pre class="r"><code>install.packages(Nbclust)</code></pre>
<p><strong>Application</strong>: The Nbclust vignette can be found at <a href="https://cran.r-project.org/web/packages/NbClust/NbClust.pdf" class="uri">https://cran.r-project.org/web/packages/NbClust/NbClust.pdf</a>.</p>
<hr />
</div>
<div id="clustvis" class="section level2">
<h2>6.2.2 <strong>ClustVis</strong></h2>
<p><strong>Introduction</strong>: This web tool allows users to upload their own data and easily create Principal Component Analysis (PCA) plots and heatmaps. Data can be uploaded as a file or by copy-pasteing it to the text box. Data format is shown under “Help” tab.</p>
<center>
<b>Data Source Statistics</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">species</td>
<td align="left">9</td>
</tr>
<tr class="even">
<td align="left">Data sources</td>
<td align="left">22</td>
</tr>
<tr class="odd">
<td align="left">Network</td>
<td align="left">7</td>
</tr>
</tbody>
</table>
<p><strong>Installation</strong>:To use the Docker image, you need to have Docker installed. Then use the following code:</p>
<pre class="bash"><code>sudo docker pull taunometsalu/clustvis
mkdir ~/customClustvis/
cd ~/customClustvis/
wget https://github.com/taunometsalu/ClustVis/archive/master.zip
unzip master.zip
chmod -R go+rx ~/customClustvis/
sudo docker run -d \
	--name customClustvis \
	-p &lt;myPort&gt;:3838 \
    -v ~/customClustvis/ClustVis-master/:/srv/shiny-server/:ro \
    taunometsalu/clustvis</code></pre>
<p>To start using ClustVis R package, you can look at the examples in the vignette that comes with the package from <a href="https://github.com/taunometsalu/ClustVis">github</a>:</p>
<pre class="r"><code>install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;pcaMethods&quot;)
library(devtools)
install_github(&quot;taunometsalu/pheatmap&quot;)
install_github(&quot;taunometsalu/clustvis/Rpackage&quot;, build_vignettes = TRUE)
vignette(&quot;vignette&quot;, &quot;clustvis&quot;)</code></pre>
<p><strong>Application</strong>: The ClustVis tutorial can be found at
<a href="https://biit.cs.ut.ee/clustvis/" class="uri">https://biit.cs.ut.ee/clustvis/</a></p>
<p><strong>Web server</strong>: <a href="https://biit.cs.ut.ee/clustvis/">ClustVis</a>.
<img src="/images/ClustVis.jpg?width=50pc" alt="ClustVis" /></p>
<hr />
</div>
<div id="medusa" class="section level2">
<h2>6.2.3 <strong>Medusa</strong></h2>
<p><strong>Introduction</strong>: Medusa3 is a java standalone application for visualization and clustering analysis of biological networks in 2D. It is highly interactive and it supports weighted and multi-edged graphs where each edge between two bioentities can represent a different biological concept. Comparing to previous versions, it is currently enriched with a variety of layout and clustering methods for more intuitive visualizations. It is easy to integrate with web application since it is offered as an applet.</p>
<center>
<b>Data Source Statistics</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">species</td>
<td align="left">9</td>
</tr>
<tr class="even">
<td align="left">Data sources</td>
<td align="left">22</td>
</tr>
<tr class="odd">
<td align="left">Network</td>
<td align="left">7</td>
</tr>
</tbody>
</table>
<p><strong>Installation</strong>:The latest version 3.0 can be downloaded directly from here.
<a href="https://docs.google.com/uc?id=0B-FSkXfyY2AFZTUxYWIxMTgtMjJiYy00NTMyLTlmMzUtZDdiZDFlZGRhYjUw&amp;export=download&amp;hl=en" class="uri">https://docs.google.com/uc?id=0B-FSkXfyY2AFZTUxYWIxMTgtMjJiYy00NTMyLTlmMzUtZDdiZDFlZGRhYjUw&amp;export=download&amp;hl=en</a></p>
<p><strong>Application</strong>: The Medusa tutorial can be found at
<a href="https://sites.google.com/site/medusa3visualization/tutorial" class="uri">https://sites.google.com/site/medusa3visualization/tutorial</a></p>
<p><strong>Web server</strong>: <a href="https://sites.google.com/site/medusa3visualization/">Medusa</a>.
<img src="/images/Medusa.jpg?width=50pc" alt="Medusa" /></p>
<hr />
</div>
<div id="wcd" class="section level2">
<h2>6.2.4 <strong>wcd</strong></h2>
<p><strong>Introduction</strong>: wcd is an open source tool for clustering expressed sequence tags. It implements matching using either d2 or edit distance, which provides sensitive matching. Heuristics speed up the clustering process. wcd parallelises well under MPI and also provides Pthreads clustering. wcd has a very compact memory representation which means that large files can be clustered on a single processor (and it can run quite happily without MPI or Pthreads headers or libraries being available).</p>
<p><strong>Application</strong>: The wcd tutorial can be found at
<a href="https://code.google.com/archive/about" class="uri">https://code.google.com/archive/about</a>.</p>
<p><strong>Web server</strong>: <a href="https://code.google.com/archive/p/wcdest/">wcd</a>.
<img src="/images/wcd.jpg?width=50pc" alt="wcd" /></p>
<hr />
</div>
<div id="timeclust" class="section level2">
<h2>6.2.5 <strong>TimeClust</strong></h2>
<p><strong>Introduction</strong>: TimeClust is an user-friendly software package to cluster genes according to their temporal expression profiles. It can be conveniently used to analyze data obtained from DNA microarray time-course experiments. It implements two original algorithms expressed designed for clustering short time series together with hierarchical clustering and self-organizing maps. TimeClust executable files for Windows and LINUX platforms can be downloaded free of charge for non-profit institutions from the following web site <a href="http://aimed11.unipv.it/TimeClust" class="uri">http://aimed11.unipv.it/TimeClust</a>.</p>
<p><strong>Installation</strong>: The TimeClust software can be downloaded freely for non-profit institutions from <a href="http://aimed11.unipv.it/TimeClust" class="uri">http://aimed11.unipv.it/TimeClust</a>. Executable files for Windows and LINUX platforms are available. The program requires the user to install the MATLAB runtime libraries, that are distributed together with the TimeClust files. The installation steps are here detailed.</p>
<p>How to install TimeClust:</p>
<ol style="list-style-type: decimal">
<li>Download the distribution file for the desired platform.</li>
<li>Extract all files from the zip file on a temporary directory.</li>
<li>Install the MATLAB libraries trough the MCRInstaller (if the same version of the MATLAB library files are not already installed) in the following way:</li>
</ol>
<ul>
<li>for Windows machines</li>
</ul>
<ol style="list-style-type: lower-alpha">
<li><p>run the MCRInstaller.exe. The setup wizard starts.</p></li>
<li><p>add the following directory to your system path: <mcr_root></p></li>
</ol>
<ul>
<li>for Unix machines</li>
</ul>
<ol style="list-style-type: lower-alpha">
<li><p>copy the MCRInstaller.zip in the installation directory and unzip the MCRInstaller.zip file.</p></li>
<li><p>update your dynamic library path as follows:</p></li>
</ol>
<p><em>Linux x86 32bit</em></p>
<pre><code> export LD_LIBRARY_PATH=&quot;&lt;mcr_root&gt;/&lt;ver&gt;/bin/glnx86:&lt;mcr_root&gt;/&lt;ver&gt;/runtime/glnx86:&lt;mcr_root&gt;/&lt;ver&gt;/sys/os/glnx86:&lt;mcr_root&gt;/&lt;ver&gt;/sys/java/jre/glnx86/jre1.5.0/lib/i386/native_threads:&lt;mcr_root&gt;/&lt;ver&gt;/sys/java/jre/glnx86/jre1.5.0/lib/i386/client:&lt;mcr_root&gt;/&lt;ver&gt;/sys/java/jre/glnx86/jre1.5.0/lib/i386:&quot;
export XAPPLRESDIR=&quot;&lt;mcr_root&gt;/&lt;ver&gt;/X11/app-defaults&quot;</code></pre>
<p><em>Linux x86-64 64bit</em></p>
<pre><code>export LD_LIBRARY_PATH=&quot;&lt;mcr_root&gt;/&lt;ver&gt;/runtime/glnxa64:&lt;mcr_root&gt;/&lt;ver&gt;/sys/os/glnxa64:&lt;mcr_root&gt;/&lt;ver&gt;/sys/java/jre/glnxa64/jre1.4.2/lib/amd64/native_threads:&lt;mcr_root&gt;/&lt;ver&gt;/sys/java/jre/glnxa64/jre1.4.2/lib/amd64/client:&lt;mcr_root&gt;/&lt;ver&gt;/sys/java/jre/glnxa64/jre1.4.2/lib/amd64:&quot;
export XAPPLRESDIR=&quot;&lt;mcr_root&gt;/&lt;ver&gt;/X11/app-defaults&quot;</code></pre>
<p>Note. If you have already installed MATLAB on your machine there is
a limitation regarding directories on your path: this tool requires
that mcr_root directory is first on the path, MATLAB requires that
the matlabroot directory is first on the path.</p>
<ol start="4" style="list-style-type: decimal">
<li><p>Copy the TimeClust executable file and the TimeClust.CTF file in
your application root directory.</p></li>
<li><p>Run for the first time the TimeClust executable file and close
the application. In this step the TimeClust_mcr directory and its
subdirectories are created.</p></li>
<li><p>Open the config.txt file in the TimeClust_mcr/TimeClust1.3
directory to configure your application. In particular, set:</p></li>
</ol>
<ul>
<li><p>at the line 2 of this file, the desired web site to open from the
TimeClust tool (default site is www.geneontology.org)</p></li>
<li><p>at the line 4, the command line to start the preferred editor to
show text files (this step is not necessary for Windows platforms,
the default text editor will be used).</p></li>
<li><p>at line 6, the command line to start the web browser to use (this
step is not necessary for Windows platforms, the default web browser
will be used).</p></li>
</ul>
<ol start="7" style="list-style-type: decimal">
<li>Run again the TimeClust executable file and try with the small
data set distributed together the software tool if all works as
reported in the user’s guide.</li>
</ol>
<p><strong>Application</strong>: The TimeClust vignette can be found at <a href="http://aimed11.unipv.it/TimeClust/example.pdf" class="uri">http://aimed11.unipv.it/TimeClust/example.pdf</a>.</p>
<hr />
</div>
<div id="hiplot" class="section level2">
<h2>6.2.6 <strong>hiplot</strong></h2>
<p><strong>Introduction</strong>: we newly developed a pool of biomedical visualization and modeling functions (240+), and proposed a scalable and emerging one-stop web service, Hiplot, with versatile task plugins, such as basic statistics, regression, clustering, genomics, transcriptomics, time series, meta-analysis, and risk models. It provides more simple web and command-line interfaces compared with other similar platforms, and is supported and maintained by the openbiox open source community.</p>
<center>
<b>Data Source Statistics</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">species</td>
<td align="left">9</td>
</tr>
<tr class="even">
<td align="left">Data sources</td>
<td align="left">22</td>
</tr>
<tr class="odd">
<td align="left">Network</td>
<td align="left">7</td>
</tr>
</tbody>
</table>
<p><strong>Application</strong>: The hiplot tutorial can be found at <a href="https://hiplot-academic.com/docs/" class="uri">https://hiplot-academic.com/docs/</a>
<strong>Web server</strong>: <a href="https://hiplot-academic.com/">hiplot</a>.
<img src="/images/hiplot.jpg?width=50pc" alt="hiplot" /></p>
<hr />
</div>
<div id="cola" class="section level2">
<h2>6.2.7 <strong>cola</strong></h2>
<p><strong>Introduction</strong>: The cola package provides a general framework for subgroup classification by consensus partitioning. It has the following features: 1. It modularizes the consensus partitioning processes that various methods can be easily integrated. 2. It provides rich visualizations for interpreting the results. 3. It allows running multiple methods at the same time and provides functionalities to straightforward compare results. 4. It provides a new method to extract features which are more efficient to separate subgroups. 5. It automatically generates detailed reports for the complete analysis. 6. It allows applying consensus partitioning in a hierarchical manner.</p>
<p><strong>Installation</strong>:To install this package, start R (version “4.2”) and enter:</p>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;cola&quot;)</code></pre>
<p><strong>Application</strong>: The cola vignette can be found at
<a href="https://bioconductor.org/packages/release/bioc/vignettes/cola/inst/doc/cola.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/cola/inst/doc/cola.html</a></p>
</div>
