---
title: "Gene and phenotype association analysis"
output:
  blogdown::html_page:
    toc: true
weight: 1
---


<div id="TOC">
<ul>
<li><a href="#predixcan" id="toc-predixcan">7.1.1 <strong>PrediXcan</strong></a></li>
<li><a href="#metaxcan" id="toc-metaxcan">7.1.2 <strong>MetaXcan</strong></a></li>
<li><a href="#fusion" id="toc-fusion">7.1.3 <strong>Fusion</strong></a></li>
<li><a href="#smr" id="toc-smr">7.1.4 <strong>SMR</strong></a></li>
<li><a href="#ptwas" id="toc-ptwas">7.1.5 <strong>PTWAS</strong></a></li>
</ul>
</div>

<p>Gene-phenotype association analysis provides valuable insights into the genetic basis of phenotypic traits and diseases, offering opportunities for biomarker discovery, personalized medicine, drug target identification, and elucidation of disease mechanisms.</p>
<p><strong>Breaking</strong>: Some user-friendly gene-phenotype association analysis tools are provided in this section.</p>
<hr />
<div id="predixcan" class="section level2">
<h2>7.1.1 <strong>PrediXcan</strong></h2>
<p><strong>Introduction</strong>: PrediXcan is a command-line tool that predicts gene expression from genotype data and performs gene-based association tests, allowing researchers to prioritize genes that are likely to be causal for a phenotype. PrediXcan consists of two steps: 1) Predict gene expression (or whatever biology the models predict) in a cohort with available genotypes; 2) Run associations to a trait measured in the cohort.</p>
<p><strong>Installation</strong>: To run PrediXcan you will need:</p>
<ul>
<li>Software Requirements:</li>
</ul>
<ol style="list-style-type: decimal">
<li>Linux or Mac OS</li>
<li>Python 2.7</li>
<li>numpy package</li>
<li>R</li>
</ol>
<ul>
<li>Scripts:</li>
</ul>
<ol style="list-style-type: decimal">
<li>PrediXcan.py</li>
<li>PrediXcanAssociation.R</li>
</ol>
<ul>
<li>Input Files:</li>
</ul>
<ol style="list-style-type: decimal">
<li>genotype file</li>
<li>sample file</li>
<li>transcriptome prediction model (sqlite db to be downloaded from PredictDB.)</li>
<li>phenotype file</li>
<li>filter file - Specifies a subset of rows on which to perform association tests (optional)</li>
</ol>
<p><strong>Application</strong>: The PrediXcan vignette can be found at <a href="https://github.com/hakyimlab/PrediXcan" class="uri">https://github.com/hakyimlab/PrediXcan</a>.</p>
<hr />
</div>
<div id="metaxcan" class="section level2">
<h2>7.1.2 <strong>MetaXcan</strong></h2>
<p><strong>Introduction</strong>: MetaXcan is a set of tools to integrate genomic information of biological mechanisms with complex traits. Almost all of the software here is command-line based. This software has been recently migrated to python 3 as python 2 has been sunset.</p>
<p><strong>Installation</strong>: The software is developed and tested in Linux and Mac OS environments. The main S-PrediXcan script is also supported in Windows.</p>
<p>To run S-PrediXcan, you need Python 3.5 or higher, with the following libraries:
numpy (&gt;=1.11.1)
scipy (&gt;=0.18.1)
pandas (&gt;=0.18.1)
sqlalchemy is needed at some unit tests.</p>
<p>To run PrediXcan Associations and MulTiXcan, you also need:
patsy (&gt;=0.5.0)
statsmodels (&gt;=0.8.0)
h5py (&gt;=2.7.1)
h5py-cache (&gt;=1.0.0) *Now folded into h5py</p>
<p>To run prediction of biological mechanisms on individual-level data, you will also need:
bgen_reader (&gt;=3.0.3)
cyvcf2 (&gt;=0.8.0)
R with ggplot and dplyr is needed for some optional statistics and charts.</p>
<p><strong>Application</strong>: The MetaXcan vignette can be found at <a href="https://github.com/hakyimlab/MetaXcan" class="uri">https://github.com/hakyimlab/MetaXcan</a>.</p>
<hr />
</div>
<div id="fusion" class="section level2">
<h2>7.1.3 <strong>Fusion</strong></h2>
<p><strong>Introduction</strong>: FUSION is a suite of tools for performing transcriptome-wide and regulome-wide association studies (TWAS and RWAS). FUSION builds predictive models of the genetic component of a functional/molecular phenotype and predicts and tests that component for association with disease using GWAS summary statistics. The goal is to identify associations between a GWAS phenotype and a functional phenotype that was only measured in reference data. We provide precomputed predictive models from multiple studies to facilitate this analysis.</p>
<p><strong>Installation</strong>:</p>
<ul>
<li>Download and unpack the FUSION software package from github:</li>
</ul>
<pre><code>wget https://github.com/gusevlab/fusion_twas/archive/master.zip
unzip master.zip
cd fusion_twas-master</code></pre>
<ul>
<li>Download and unpack the (1000 Genomes) LD reference data:</li>
</ul>
<pre><code>wget https://data.broadinstitute.org/alkesgroup/FUSION/LDREF.tar.bz2
tar xjvf LDREF.tar.bz2</code></pre>
<ul>
<li>Download and unpack the plink2R library (by Gad Abraham):</li>
</ul>
<pre><code>wget https://github.com/gabraham/plink2R/archive/master.zip
unzip master.zip</code></pre>
<ul>
<li>Launch R and install required libraries:</li>
</ul>
<pre class="r"><code>install.packages(c(&#39;optparse&#39;,&#39;RColorBrewer&#39;))
install.packages(&#39;plink2R-master/plink2R/&#39;,repos = NULL)</code></pre>
<p>If computing your own weights, the following additional steps are required
- Add the bundled GCTA binary <code>gcta_nr_robust</code> to path (coded by Po-Ru Loh for robust non-linear optimization)
- Download and install PLINK2, add <code>plink</code> to path
- Launch R and install the following required libraries:</p>
<pre class="r"><code>install.packages(c(&#39;glmnet&#39;,&#39;methods&#39;))</code></pre>
<ul>
<li>If using BSLMM, download and install GEMMA software, add to path. Generate a symbolic link to the output by calling <code>ln -s ./ output</code> in the directory where you will run <code>FUSION.weights.R</code> (this is a workaround because GEMMA requires results to go into an <code>output</code> subdirectory).</li>
</ul>
<p><strong>Application</strong>: The Fusion vignette can be found at <a href="http://gusevlab.org/projects/fusion/" class="uri">http://gusevlab.org/projects/fusion/</a>.</p>
<hr />
</div>
<div id="smr" class="section level2">
<h2>7.1.4 <strong>SMR</strong></h2>
<p><strong>Introduction</strong>: The SMR software tool was originally developed to implement the SMR &amp; HEIDI methods to test for pleiotropic association between the expression level of a gene and a complex trait of interest using summary-level data from GWAS and expression quantitative trait loci (eQTL) studies (Zhu et al. 2016 Nature Genetics). The SMR &amp; HEIDI methodology can be interpreted as an analysis to test if the effect size of a SNP on the phenotype is mediated by gene expression. This tool can therefore be used to prioritize genes underlying GWAS hits for follow-up functional studies. The methods are applicable to all kinds of molecular QTL (xQTL) data, including DNA methylation QTL (mQTL) and protein abundance QTL (pQTL).</p>
<p><strong>Installation</strong>: The source code are released under GPL v2.</p>
<p>Source code: smr_v1.3.1_src.tar.gz <a href="https://yanglab.westlake.edu.cn/software/smr/download/smr_v1.3.1_src.tar.gz" class="uri">https://yanglab.westlake.edu.cn/software/smr/download/smr_v1.3.1_src.tar.gz</a>.</p>
<p><strong>Application</strong>: The SMR vignette can be found at <a href="https://yanglab.westlake.edu.cn/software/smr/#Overview" class="uri">https://yanglab.westlake.edu.cn/software/smr/#Overview</a>.</p>
<hr />
</div>
<div id="ptwas" class="section level2">
<h2>7.1.5 <strong>PTWAS</strong></h2>
<p><strong>Introduction</strong>: A computational framework, probabilistic transcriptome-wide association study (PTWAS) can investigate causal relationships between gene expressions and complex traits. PTWAS applies the established principles from instrumental variables analysis and takes advantage of probabilistic eQTL annotations to delineate and tackle the unique challenges arising in TWAS. PTWAS not only confers higher power than the existing methods but also provides novel functionalities to evaluate the causal assumptions and estimate tissue- or cell-type-specific gene-to-trait effects.</p>
<p><strong>Installation</strong>: Software requirement</p>
<p>The bare minimum set of software packages for running PTWAS scan, validation, and estimation procedures are:</p>
<ul>
<li>GAMBIT: software package implementing generalized burden test using GWAS summary statistics (<a href="https://github.com/corbinq/GAMBIT" class="uri">https://github.com/corbinq/GAMBIT</a>).</li>
<li>PTWAS_est: perl script for causal assumption validation and effect estimation (<a href="https://github.com/xqwen/ptwas/tree/master/PTWAS_est/" class="uri">https://github.com/xqwen/ptwas/tree/master/PTWAS_est/</a>).</li>
<li>tabix: <a href="https://github.com/samtools/tabix" class="uri">https://github.com/samtools/tabix</a>.</li>
</ul>
<p>To compute PTWAS composite instrumental variables/eQTL weights for scan analysis from users’ own eQTL data, it requires additional software packages/libraries</p>
<ul>
<li>DAP: Bayesian multi-SNP analysis software for fine-mapping eQTLs (<a href="https://github.com/xqwen/dap" class="uri">https://github.com/xqwen/dap</a>).</li>
<li>ptwas_builder: utility to extract information from DAP output (part of the DAP distribution): <a href="https://github.com/xqwen/dap/tree/master/ptwas_builder" class="uri">https://github.com/xqwen/dap/tree/master/ptwas_builder</a>.</li>
<li>make_GAMBIT_DB.R: R script to format GAMBIT weights file. Requires data.table library <a href="https://github.com/xqwen/ptwas/tree/master/PTWAS_scan/" class="uri">https://github.com/xqwen/ptwas/tree/master/PTWAS_scan/</a>.</li>
</ul>
<p><strong>Application</strong>: The PTWAS vignette can be found at <a href="https://xqwen.github.io/ptwas/" class="uri">https://xqwen.github.io/ptwas/</a>.</p>
</div>
