<!DOCTYPE html>
<html lang="en" class="js csstransforms3d">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="generator" content="Hugo 0.101.0" />
    <meta name="description" content="Gene list">
<meta name="author" content="Huachun Yin">

    <link rel="icon" href="../../images/favicon.png" type="image/png">

    <title>Discovering the cell markers - Gene list mining</title>

    
    <link href="../../css/nucleus.css?1672904510" rel="stylesheet">
    <link href="../../css/fontawesome-all.min.css?1672904510" rel="stylesheet">
    <link href="../../css/hybrid.css?1672904510" rel="stylesheet">
    <link href="../../css/featherlight.min.css?1672904510" rel="stylesheet">
    <link href="../../css/perfect-scrollbar.min.css?1672904510" rel="stylesheet">
    <link href="../../css/auto-complete.css?1672904510" rel="stylesheet">
    <link href="../../css/atom-one-dark-reasonable.css?1672904510" rel="stylesheet">
    <link href="../../css/theme.css?1672904510" rel="stylesheet">
    <link href="../../css/tabs.css?1672904510" rel="stylesheet">
    <link href="../../css/hugo-theme.css?1672904510" rel="stylesheet">
    
    <link href="../../css/theme-mine.css?1672904510" rel="stylesheet">
    
    <link href="../../css/foo.css?1672904510" rel="stylesheet"><link href="../../css/bar.css?1672904510" rel="stylesheet">

    <script src="../../js/jquery-3.3.1.min.js?1672904510"></script>

    <style>
      :root #header + #content > #left > #rlblock_left{
          display:none !important;
      }
      
    </style>
    
  </head>
  <body class="" data-url="../../conditional-analysis/cellcomposition/">
    <nav id="sidebar" class="">



  <div id="header-wrapper">
    <div id="header">
      <a id="logo" href="../../">
    <img id="DGEs" src="../../images/logo.png" width="162" height="185" class="zoom">
</a>

    </div>
    
        <div class="searchbox">
    <label for="search-by"><i class="fas fa-search"></i></label>
    <input data-search-input id="search-by" type="search" placeholder="Search...">
    <span data-search-clear=""><i class="fas fa-times"></i></span>
</div>

<script type="text/javascript" src="../../js/lunr.min.js?1672904510"></script>
<script type="text/javascript" src="../../js/auto-complete.js?1672904510"></script>
<script type="text/javascript">
    
        var baseurl = "\/";
    
</script>
<script type="text/javascript" src="../../js/search.js?1672904510"></script>

    
  </div>
  

    <div class="highlightable">
    <ul class="topics">

        
          
          




 
  
    
    <li data-nav-id="/gene-annotation/" title="Gene Function Annotations" class="dd-item
        
        
        
        ">
      <a href="../../gene-annotation/">
          Gene Function Annotations
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/gene-annotation/orth/" title="Orthologous gene mapping across species" class="dd-item ">
        <a href="../../gene-annotation/orth/">
        Orthologous gene mapping across species
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/gene-annotation/goananotation/" title="Gene ontology annotation " class="dd-item ">
        <a href="../../gene-annotation/goananotation/">
        Gene ontology annotation 
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/gene-annotation/biopathway/" title="Biological pathway annotation" class="dd-item ">
        <a href="../../gene-annotation/biopathway/">
        Biological pathway annotation
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/gene-annotation/other_gene_annotation/" title="Other gene annotation" class="dd-item ">
        <a href="../../gene-annotation/other_gene_annotation/">
        Other gene annotation
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
          




 
  
    
    <li data-nav-id="/network-analysis/" title="Network and Module Analysis" class="dd-item
        
        
        
        ">
      <a href="../../network-analysis/">
          Network and Module Analysis
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/network-analysis/gene_network/" title="Gene regulatory network analysis" class="dd-item ">
        <a href="../../network-analysis/gene_network/">
        Gene regulatory network analysis
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/network-analysis/tf_regulation/" title="TF gene interaction database" class="dd-item ">
        <a href="../../network-analysis/tf_regulation/">
        TF gene interaction database
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/network-analysis/mirna_regulation/" title="miRNA-gene interaction database" class="dd-item ">
        <a href="../../network-analysis/mirna_regulation/">
        miRNA-gene interaction database
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/network-analysis/cerna_regulation/" title="ceRNA-gene interaction database" class="dd-item ">
        <a href="../../network-analysis/cerna_regulation/">
        ceRNA-gene interaction database
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/network-analysis/analysis_visualization/" title="Analysis and Visualization Tools" class="dd-item ">
        <a href="../../network-analysis/analysis_visualization/">
        Analysis and Visualization Tools
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
          




 
  
    
    <li data-nav-id="/text-mining/" title="Text Mining" class="dd-item
        
        
        
        ">
      <a href="../../text-mining/">
          Text Mining
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/text-mining/text_mining/" title="Knowledge Graph/text mining" class="dd-item ">
        <a href="../../text-mining/text_mining/">
        Knowledge Graph/text mining
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
          




 
  
    
    <li data-nav-id="/parmocogenomics/" title="Prediction of Potential Drugs" class="dd-item
        
        
        
        ">
      <a href="../../parmocogenomics/">
          Prediction of Potential Drugs
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/parmocogenomics/drug/" title="Prediction of Potential Drug (Pharmocogenomics)" class="dd-item ">
        <a href="../../parmocogenomics/drug/">
        Prediction of Potential Drug (Pharmocogenomics)
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
          




 
  
    
    <li data-nav-id="/conditional-analysis/" title="Condition-specific Analysis" class="dd-item
        parent
        
        
        ">
      <a href="../../conditional-analysis/">
          Condition-specific Analysis
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/conditional-analysis/spatialandtemporal/" title="Spatial and temporal variation" class="dd-item ">
        <a href="../../conditional-analysis/spatialandtemporal/">
        Spatial and temporal variation
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/conditional-analysis/cellcomposition/" title="Discovering the cell markers" class="dd-item active">
        <a href="../../conditional-analysis/cellcomposition/">
        Discovering the cell markers
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/conditional-analysis/trajectory/" title="Tajectory inference" class="dd-item ">
        <a href="../../conditional-analysis/trajectory/">
        Tajectory inference
        
        </a>
    </li>
     
  
 

            
          
            
            




 
  
    
      <li data-nav-id="/conditional-analysis/cellcomunication/" title="Cell communication" class="dd-item ">
        <a href="../../conditional-analysis/cellcomunication/">
        Cell communication
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
          




 
  
    
    <li data-nav-id="/clustering/" title="Clustering Analysis" class="dd-item
        
        
        
        ">
      <a href="../../clustering/">
          Clustering Analysis
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/clustering/clustering/" title="Clustering Analysis" class="dd-item ">
        <a href="../../clustering/clustering/">
        Clustering Analysis
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
          




 
  
    
    <li data-nav-id="/survival-analysis/" title="Survival Analysis" class="dd-item
        
        
        
        ">
      <a href="../../survival-analysis/">
          Survival Analysis
          
      </a>
      
      
        <ul>
          
          
          

        
          
            
            




 
  
    
      <li data-nav-id="/survival-analysis/survival_analysis/" title="Survival Analysis" class="dd-item ">
        <a href="../../survival-analysis/survival_analysis/">
        Survival Analysis
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
      
    </li>
  
 

          
        
    </ul>

    
    

    
    <section id="footer">
      <p>
  Built by <a href="https://github.com/HuaChunY?tab=repositories">Huachun Yin</a> and <a href="https://github.com/duohongrui?tab=repositories">Hongrui Duo</a>
</p>

    </section>
  </div>
</nav>




        <section id="body">
        <div id="overlay"></div>
        <div class="padding highlightable">
              
              <div>
                <div id="top-bar">
                
                
                <div id="breadcrumbs" itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb">
                    <span id="sidebar-toggle-span">
                        <a href="#" id="sidebar-toggle" data-sidebar-toggle="">
                          <i class="fas fa-bars"></i>
                        </a>
                    </span>
                  
                  <span id="toc-menu"><i class="fas fa-list-alt"></i></span>
                  
                  <span class="links">
                 
                 
                    
          
          
            
            
          
          
            
            
          
          
            <a href='../../'></a> > <a href='../../conditional-analysis/'>Condition-specific Analysis</a> > Discovering the cell markers
          
        
          
        
          
        
                 
                  </span>
                </div>
                
                    <div class="progress">
    <div class="wrapper">

    </div>
</div>

                
              </div>
            </div>
            
        <div id="head-tags">
        
        </div>
        
        <div id="body-inner">
          
            <h1>
              
              Discovering the cell markers
            </h1>
          

        




<div id="TOC">
<ul>
<li><a href="#ciersort" id="toc-ciersort">5.2.1 <strong>CIERSORT</strong></a></li>
<li><a href="#estimate" id="toc-estimate">5.2.2 <strong>ESTIMATE</strong></a></li>
<li><a href="#xcell" id="toc-xcell">5.2.2 <strong>xCell</strong></a></li>
<li><a href="#timer" id="toc-timer">5.2.3 <strong>TIMER</strong></a></li>
<li><a href="#mcpcounter" id="toc-mcpcounter">5.2.4 <strong>MCPcounter</strong></a></li>
<li><a href="#bseq_sc" id="toc-bseq_sc">5.2.5 <strong>Bseq_SC</strong></a></li>
<li><a href="#epic" id="toc-epic">5.2.6 <strong>EPIC</strong></a></li>
<li><a href="#quantiseq" id="toc-quantiseq">5.2.7 <strong>quanTIseq</strong></a></li>
<li><a href="#abbas-et-al." id="toc-abbas-et-al.">5.2.8 <strong>Abbas <em>et al.</em> </strong></a></li>
<li><a href="#abis" id="toc-abis">5.2.7 <strong>ABIS</strong></a></li>
<li><a href="#immucellai" id="toc-immucellai">5.2.8 <strong>ImmuCellAI</strong></a></li>
<li><a href="#music" id="toc-music">5.2.9 <strong>MuSiC</strong></a></li>
<li><a href="#deconrnaseq" id="toc-deconrnaseq">5.2.10 <strong>DeconRNASeq</strong></a></li>
<li><a href="#max-schelker-et-al." id="toc-max-schelker-et-al.">5.2.11 <strong>Max Schelker <em>et al.</em> </strong></a></li>
<li><a href="#dsa" id="toc-dsa">5.2.12 <strong>DSA</strong></a></li>
<li><a href="#singscore" id="toc-singscore">5.2.13 <strong>singscore</strong></a></li>
<li><a href="#ting-gong-et-al." id="toc-ting-gong-et-al.">5.2.13 <strong>Ting Gong <em>et al.</em> </strong></a></li>
<li><a href="#pert" id="toc-pert">5.2.14 <strong>PERT</strong></a></li>
<li><a href="#immucc" id="toc-immucc">5.2.14 <strong>ImmuCC</strong></a></li>
<li><a href="#immunostates" id="toc-immunostates">5.2.15 <strong>immunoStates</strong></a></li>
<li><a href="#dcq" id="toc-dcq">5.2.16 <strong>DCQ</strong></a></li>
<li><a href="#scdc" id="toc-scdc">5.2.17 <strong>SCDC</strong></a></li>
<li><a href="#edec" id="toc-edec">5.2.18 <strong>Edec</strong></a></li>
<li><a href="#mmad" id="toc-mmad">5.2.19 <strong>MMAD</strong></a></li>
<li><a href="#scbio" id="toc-scbio">5.2.20 <strong>scBio</strong></a></li>
<li><a href="#debcam" id="toc-debcam">5.2.21 <strong>debCAM</strong></a></li>
<li><a href="#temt" id="toc-temt">5.2.22 <strong>TEMT</strong></a></li>
<li><a href="#demixt" id="toc-demixt">5.2.23 <strong>DeMixT</strong></a></li>
<li><a href="#dtangle" id="toc-dtangle">5.2.24 <strong>Dtangle</strong></a></li>
<li><a href="#fardeep" id="toc-fardeep">5.2.25 <strong>FARDEEP</strong></a></li>
<li><a href="#toast" id="toc-toast">5.2.26 <strong>TOAST</strong></a></li>
<li><a href="#cdseq" id="toc-cdseq">5.2.27 <strong>CDSeq</strong></a></li>
<li><a href="#casey-p.-shannon-et-al." id="toc-casey-p.-shannon-et-al.">5.2.28 <strong>Casey P. Shannon <em>et al.</em> </strong></a></li>
<li><a href="#deconvseq" id="toc-deconvseq">5.2.29 <strong>deconvSeq</strong></a></li>
<li><a href="#xiaoqing-yu-et-al." id="toc-xiaoqing-yu-et-al.">5.2.30 <strong>Xiaoqing Yu <em>et al.</em> </strong></a></li>
<li><a href="#digitaldlsorter" id="toc-digitaldlsorter">5.2.31 <strong>Digitaldlsorter</strong></a></li>
<li><a href="#declust" id="toc-declust">5.2.32 <strong>DeClust</strong></a></li>
<li><a href="#mysort" id="toc-mysort">5.2.33 <strong>MySort</strong></a></li>
<li><a href="#smcgenedeconv" id="toc-smcgenedeconv">5.2.34 <strong>smcgenedeconv</strong></a></li>
<li><a href="#chiu-yen-jung-et-al." id="toc-chiu-yen-jung-et-al.">5.2.35 <strong>Chiu Yen-Jung <em>et al.</em> </strong></a></li>
<li><a href="#nitumid" id="toc-nitumid">5.2.36 <strong>NITUMID</strong></a></li>
<li><a href="#adapts" id="toc-adapts">5.2.37 <strong>ADAPTS</strong></a></li>
<li><a href="#momf" id="toc-momf">5.2.38 <strong>MOMF</strong></a></li>
<li><a href="#deblender" id="toc-deblender">5.2.39 <strong>Deblender</strong></a></li>
<li><a href="#lishuang-qi-et-al." id="toc-lishuang-qi-et-al.">5.2.40 <strong>Lishuang Qi <em>et al.</em> </strong></a></li>
<li><a href="#mixture" id="toc-mixture">5.2.41 <strong>MIXTURE</strong></a></li>
<li><a href="#semi-cam" id="toc-semi-cam">5.2.42 <strong>Semi-CAM</strong></a></li>
<li><a href="#dtd" id="toc-dtd">5.2.43 <strong>DTD</strong></a></li>
<li><a href="#infino" id="toc-infino">5.2.44 <strong>Infino</strong></a></li>
</ul>
</div>

<hr />
<div id="ciersort" class="section level2">
<h2>5.2.1 <strong>CIERSORT</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: CIBERSORT( or CIBERSORTx) is a gene expression-based deconvolution algorithm that provides an estimation of the abundances of member cell types in a mixed cell population using gene expression data. CIBERSORTx allows users to process gene expression data representing a bulk admixture of different cell types, along with a signature matrix file that enumerates the genes defining the expression profile for each cell type of interest. For the latter, users can either use exissting/curated signature matrices for reference cell types, or can create custom signature gene files by providing the reference gene expression profiles o of pure cell populations. Moreover, given the increasing use of single cell transcriptome sequencing, CIBERSORTx also provides the option to derive signature matrices from sinqle-cell RNA sequencing data.</p></li>
<li><p><strong>Web server</strong>: <a href="https://cibersort.stanford.edu/index.php">CIERSORT</a></p></li>
</ul>
<hr />
</div>
<div id="estimate" class="section level2">
<h2>5.2.2 <strong>ESTIMATE</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: ESTIMATE (Estimation of STromal and Immune cells in MAlignant Tumor tissues using Expression data) provides researchers with scores for tumor purity, the level of stromal cells present, and the infiltration level of immune cells in tumor tissues based on expression data. This website is designed to view and download stromal, immune, and ESTIMATE scores for each sample across all TCGA (The Cancer Genome Atlas) tumor types and platforms. ESTIMATE algorithm is based on single sample Gene Set Enrichment Analysis and generates three scores:
1.stromal score (that captures the presence of stroma in tumor tissue),
2.immune score (that represents the infiltration of immune cells in tumor tissue), and
3.estimate score (that infers tumor purity).</p></li>
<li><p><strong>Installation</strong>: In early October 2016, the ESTIMATE R package was migrated to R-Forge:</p></li>
</ul>
<pre class="r"><code>library(utils)
rforge &lt;- &quot;http://r-forge.r-project.org&quot;
install.packages(&quot;estimate&quot;, repos=rforge, dependencies=TRUE)</code></pre>
<ul>
<li><strong>Web server</strong>: <a href="https://bioinformatics.mdanderson.org/estimate/">ESTIMATE</a></li>
</ul>
<hr />
</div>
<div id="xcell" class="section level2">
<h2>5.2.2 <strong>xCell</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: xCell is a webtool that performs cell type enrichment analysis from gene expression data for 64 immune and stroma cell types. xCell is a gene signatures-based method learned from thousands of pure cell types from various sources. xCell applies a novel technique for reducing associations between closely related cell types. xCell signatures were validated using extensive in-silico simulations and also cytometry immunophenotyping, and were shown to outperform previous methods. xCell allows researchers to reliably portray the cellular heterogeneity landscape of tissue expression profiles.</p></li>
<li><p><strong>Installation</strong>: To install it, the easiest way is to use the R package “devtools” and its function “install_github”:</p></li>
</ul>
<pre class="r"><code>devtools::install_github(&#39;dviraran/xCell&#39;)</code></pre>
<ul>
<li><strong>Web server</strong>: <a href="http://xCell.ucsf.edu/">xCell</a></li>
</ul>
<hr />
</div>
<div id="timer" class="section level2">
<h2>5.2.3 <strong>TIMER</strong></h2>
<ul>
<li><strong>Introduction</strong>: TIMER web server is a comprehensive resource for systematical analysis of immune infiltrates across diverse cancer types. The abundances of six immune infiltrates (B cells, CD4+ T cells, CD8+ T cells, Neutrophils, Macrophages, and Dendritic cells) are estimated by TIMER algorithm. TIMER web server allows users to input function-specific parameters, with resulting figures dynamically displayed to conveniently access the tumor immunological, clinical, and genomic features.</li>
</ul>
<p>+Go Gene module to explore the correlation between gene expression and abundance of immune infiltrates;
+Go Survival module to explore the association between clinical outcome and abundance of immune infiltrates or gene expression;
+Go Mutation module to explore the correlation between mutated genes and abundance of immune infiltrates;
+Go SCNA module to explore the correlation between somatic CNA and abundance of immune infiltrates;
+Go Diff Exp module to explore differential gene expression between tumor and normal tissue;
+Go Correlation module to explore correlations between genes ;
+Go Estimation module to run users’ private samples by TIMER algorithm.</p>
<ul>
<li><strong>Web server</strong>: <a href="https://cistrome.shinyapps.io/timer/">TIMER</a></li>
</ul>
<hr />
</div>
<div id="mcpcounter" class="section level2">
<h2>5.2.4 <strong>MCPcounter</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: MCP-counter (Microenvironment Cell Populations-counter) method, which allows the robust quantification of the absolute abundance of eight immune and two stromal cell populations in heterogeneous tissues from transcriptomic data.</p></li>
<li><p><strong>Installation</strong>: To install it, the easiest is to use the “R” package “devtools” and its function “install_github”. To do so, open an R session and enter:</p></li>
</ul>
<pre class="r"><code>devtools::install_github(&#39;dviraran/xCell&#39;)</code></pre>
<hr />
</div>
<div id="bseq_sc" class="section level2">
<h2>5.2.5 <strong>Bseq_SC</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: BSeq-sc is a bioinformatics analysis pipeline that leverages single-cell sequencing data to estimate cell type proportion and cell type-specific gene expression differences from
RNA-seq data from bulk tissue samples.</p></li>
<li><p><strong>Installation</strong>: To install it, the easiest is to use the “R” package “devtools” and its function “install_github”. To do so, open an R session and enter:</p></li>
</ul>
<pre class="r"><code># install devtools if necessary
install.packages(&#39;devtools&#39;)

# install the bseqsc package
devtools::install_github(&#39;shenorrlab/bseq-sc&#39;)

# load
library(bseqsc)</code></pre>
<hr />
</div>
<div id="epic" class="section level2">
<h2>5.2.6 <strong>EPIC</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: EPIC application is designed to Estimate the Proportion of Immune and Cancer cells from bulk tumor gene expression data. This is done by fitting gene expression reference profiles from the main non-malignant cell types and simultaneously accounting for an uncharacterized cell type without prior knowledge about it (e.g. cancer cells in solid tumors samples). the reference gene expression profiles derived from the main tumor infiltrating cell types (i.e. immune subsets, stromal and endothelial cells). This method could however also be applied to predict the cell fractions in other types of mixed samples if reference gene expression profiles from these other cell types are available. The main function in this package is EPIC. It needs as input a matrix of the TPM (or RPKM) gene expression from the samples for which to estimate cell proportions.</p></li>
<li><p><strong>Installation</strong>: To install it, the easiest is to use the “R” package “devtools” and its function “install_github”. To do so, open an R session and enter:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;GfellerLab/EPIC&quot;, build_vignettes=TRUE)</code></pre>
<p>A pyhton wrapper has been written by Stephen C. Van Nostrand from MIT and is available at <a href="https://github.com/scvannost/epicpy" class="uri">https://github.com/scvannost/epicpy</a>.</p>
<ul>
<li><strong>Web server</strong>: <a href="http://epic.gfellerlab.org/">EPIC</a></li>
</ul>
<hr />
</div>
<div id="quantiseq" class="section level2">
<h2>5.2.7 <strong>quanTIseq</strong></h2>
<p><img src="../../images/quanTIseq.png?width=15pc" /></p>
<ul>
<li><p><strong>Introduction</strong>: quanTIseq is a computational pipeline for the quantification of the Tumor Immune contexture from human RNA-seq data. quanTIseq takes as input FASTQ files of RNA-seq reads from tumor samples or other cell mixtures and quantifies via deconvolution the proportions of ten different immune cell types and the fraction of other uncharacterized cells present in the heterogeneous sample.
quanTIseq analysis consists of three steps (Figure 1):
1.Pre-processing of raw RNA-seq reads (single- or paired-ends) with Trimmomatic to remove Illumina adapter sequences, trim low-quality read ends, crop long-reads to a maximum length, and discard short reads.
2.Quantification of gene expression with Kallisto [3] as transcripts per millions (TPM) and raw counts.
3.Expression normalization, gene re-annotation, deconvolution of cell fractions based on constrained least squares regression, and computation of cell densities (if total cells per mm<sup>2</sup> are available from images of tumor tissue-slides).</p></li>
<li><p><strong>Installation</strong>: 1.Mac OS X (based on Docker): install Docker (<a href="https://docs.docker.com/engine/installation/">instructions here</a>) and download the “quanTIseq_pipeline.sh” script from <a href="https://github.com/icbi-lab/quanTIseq/raw/master/quanTIseq_pipeline.sh">here</a>.
2.Linux(based on Singularity): install Singularity (<a href="https://www.sylabs.io/guides/3.0/user-guide/">instructions here</a>) and download the “quanTIseq_pipeline.sh” script from <a href="https://github.com/icbi-lab/quanTIseq/raw/master/quanTIseq_pipeline.sh">here</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="https://icbi.i-med.ac.at/software/quantiseq/doc/">quanTIseq</a></p></li>
</ul>
<hr />
</div>
<div id="abbas-et-al." class="section level2">
<h2>5.2.8 <strong>Abbas <em>et al.</em> </strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Abbas <em>et al.</em> demonstrate that microarray expression deconvolution accurately quantifies the constituents of real blood samples and mixtures of immune-derived cell lines.</p></li>
<li><p><strong>Installation</strong>: The formula can be found here: <a href="https://doi.org/10.1371/journal.pone.0006098" class="uri">https://doi.org/10.1371/journal.pone.0006098</a></p></li>
</ul>
<hr />
</div>
<div id="abis" class="section level2">
<h2>5.2.7 <strong>ABIS</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: This Shiny app performs absolute deconvolution on RNA-Seq and microarray data. It also contain a Gene Viewer page where the expression of a gene can be visualized across 29 immune cell types.</p></li>
<li><p><strong>Installation</strong>: You need to download the app from GitHub through R and it will run locally. However, as soon as you will close R, the app will not be available anymore and you need it to download it again. All the packages and dependencies have to be installed first.</p></li>
</ul>
<pre class="r"><code>install.packages(c(&quot;shiny&quot;, &quot;MASS&quot;, &quot;preprocessCore&quot;), dependencies = TRUE)

shiny::runGitHub(&quot;ABIS&quot;, user=&quot;giannimonaco&quot;)</code></pre>
<ul>
<li><strong>Web server</strong>: <a href="https://giannimonaco.shinyapps.io/ABIS/">ABIS</a></li>
</ul>
<hr />
</div>
<div id="immucellai" class="section level2">
<h2>5.2.8 <strong>ImmuCellAI</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: ImmuCellAI (Immune Cell Abundance Identifier) is a tool to estimate the abundance of 24 immune cells from gene expression dataset including RNA-Seq and microarray data, in which the 24 immune cells are comprised of 18 T-cell subtypes and 6 other immune cells: B cell, NK cell, Monocyte cell, Macrophage cell, Neutrophil cell and DC cell. Besides, ImmuCellAI can be applied to estimate the difference of immune cell infiltration in divers groups as well as predict patients response to immune checkpoint blockade therapy.</p></li>
<li><p><strong>Installation</strong>: To install it, the easiest is to use the “R” package “devtools” and its function “install_github”. To do so, open an R session and enter:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
library(devtools)
install_github(&quot;lydiaMyr/ImmuCellAI@main&quot;)
#if the &quot;/bin/gtar: not found&quot; error occured, please run the following command &quot;export TAR=&quot;/bin/tar&quot; before installation.</code></pre>
<p>The R package of ImmuCellAI-mouse:</p>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
library(devtools)
install_github(&quot;lydiaMyr/ImmuCellAI-mouse@main&quot;)
#if the &quot;/bin/gtar: not found&quot; error occured, please run the following command &quot;export TAR=&quot;/bin/tar&quot; before installation.</code></pre>
<ul>
<li><strong>Web server</strong>: <a href="http://bioinfo.life.hust.edu.cn/ImmuCellAI#!/">ImmuCellAI</a>
<img src="../../images/ImmuCellAI.jpg?width=50pc" alt="ImmuCellAI" /></li>
</ul>
<hr />
</div>
<div id="music" class="section level2">
<h2>5.2.9 <strong>MuSiC</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: MuSiC is an analysis toolkit for single-cell RNA-Seq experiments. To use this package, you will need the R statistical computing environment (version 3.0 or later) and several packages available through Bioconductor and CRAN.</p></li>
<li><p><strong>Installation</strong>: To install it, the easiest is to use the “R” package “devtools” and its function “install_github”. To do so, open an R session and enter:</p></li>
</ul>
<pre class="r"><code># install devtools if necessary
install.packages(&#39;devtools&#39;)

# install the MuSiC package
devtools::install_github(&#39;xuranw/MuSiC&#39;)

# load
library(MuSiC)</code></pre>
<hr />
</div>
<div id="deconrnaseq" class="section level2">
<h2>5.2.10 <strong>DeconRNASeq</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: DeconSeq is an R package for deconvolution of heterogeneous tissues based on mRNA-Seq data. It modeled expression levels from heterogeneous cell populations in mRNA-Seq as the weighted average of expression from different constituting cell types and predicted cell type proportions of single expression profiles.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version “4.2”) and enter:</p></li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;DeconRNASeq&quot;)</code></pre>
<hr />
</div>
<div id="max-schelker-et-al." class="section level2">
<h2>5.2.11 <strong>Max Schelker <em>et al.</em> </strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Max Schelker <em>et al.</em> demonstrate how to derive the cellular composition of a solid tumour from bulk gene expression data by mathematical deconvolution, using indication-specific and cell type-specific reference gene expression profiles (RGEPs) from tumour-derived single-cell RNA sequencing data.</p></li>
<li><p><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://figshare.com/s/711d3fb2bd3288c8483a" class="uri">https://figshare.com/s/711d3fb2bd3288c8483a</a></p></li>
</ul>
<hr />
</div>
<div id="dsa" class="section level2">
<h2>5.2.12 <strong>DSA</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: DSA (Digital Sorting Algorithm) is a algorithm for extracting cell-type specific gene expression profiles from mixed tissue samples that is unbiased and does not require prior knowledge of cell type frequencies.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;DSA&quot;)</code></pre>
<hr />
</div>
<div id="singscore" class="section level2">
<h2>5.2.13 <strong>singscore</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A simple single-sample gene signature scoring method that uses rank-based statistics to analyze the sample’s gene expression profile. It scores the expression activities of gene sets at a single-sample level.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p></li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;singscore&quot;)</code></pre>
<hr />
</div>
<div id="ting-gong-et-al." class="section level2">
<h2>5.2.13 <strong>Ting Gong <em>et al.</em> </strong></h2>
<ul>
<li><p><strong>Introduction</strong>: An approach that builds upon a linear latent variable model where expression levels from mixed cell populations are modeled as the weighted average of expression from different cell types.</p></li>
<li><p><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://doi.org/10.1371/journal.pone.0027156" class="uri">https://doi.org/10.1371/journal.pone.0027156</a></p></li>
</ul>
<hr />
</div>
<div id="pert" class="section level2">
<h2>5.2.14 <strong>PERT</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A method for expression deconvolution of human blood samples from varied microenvironmental and developmental conditions.</p></li>
<li><p><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://doi.org/10.1371/journal.pcbi.1002838.s003" class="uri">https://doi.org/10.1371/journal.pcbi.1002838.s003</a></p></li>
</ul>
<hr />
</div>
<div id="immucc" class="section level2">
<h2>5.2.14 <strong>ImmuCC</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Four files including “ImmuCellDB”,“tissue_immucc”,“seq_ImmuCC” and “array_ImmuCC” were uesed to analyze cell populations.</p></li>
<li><p><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://doi.org/10.1371/journal.pcbi.1002838.s003" class="uri">https://doi.org/10.1371/journal.pcbi.1002838.s003</a></p></li>
</ul>
<hr />
</div>
<div id="immunostates" class="section level2">
<h2>5.2.15 <strong>immunoStates</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A basis matrix built using 6160 samples with different disease states across 42 microarray platforms. The immunoStates significantly reduces biological and technical biases.</p></li>
<li><p><strong>Installation</strong>: The deconvolution “R” code can be found here: <a href="https://doi.org/10.1371/journal.pcbi.1002838.s003" class="uri">https://doi.org/10.1371/journal.pcbi.1002838.s003</a></p></li>
</ul>
<hr />
</div>
<div id="dcq" class="section level2">
<h2>5.2.16 <strong>DCQ</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: DCQ application is operational on different browsers including FireFox version 22
(22.0 and higher), Chrome version 28 (28.0.1500.72 and higher), Safari version 6
(6.0.2 and higher) and Explorer version 10 (10.0.9200 and higher).</p></li>
<li><p><strong>Installation</strong>: A R package named ComICS is now available:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;ComICS&quot;)</code></pre>
<ul>
<li><strong>Web server</strong>: <a href="http://dcq.tau.ac.il/">DCQ</a></li>
</ul>
<hr />
</div>
<div id="scdc" class="section level2">
<h2>5.2.17 <strong>SCDC</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: SCDC is a deconvolution method for bulk RNA-seq that leverages cell-type specific gene expressions from multiple scRNA-seq reference datasets. SCDC adopts an ENSEMBLE method to integrate deconvolution results from different scRNA-seq datasets that are produced in different laboratories and at different times, implicitly addressing the batch-effect confounding.</p></li>
<li><p><strong>Installation</strong>: You can install the released version of SCDC from GitHub with:</p></li>
</ul>
<pre class="r"><code>if (!require(&quot;devtools&quot;)) {
  install.packages(&quot;devtools&quot;)
}
devtools::install_github(&quot;meichendong/SCDC&quot;)</code></pre>
<p>Dependency package problem regarding to ‘xbioc’ could be resolved by:</p>
<pre class="r"><code>install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;renozao/xbioc&quot;)</code></pre>
<ul>
<li><strong>Web server</strong>: <a href="https://meichendong.github.io/SCDC/index.html">SCDC</a>
<img src="../../images/SCDC.jpg?width=50pc" alt="SCDC" /></li>
</ul>
<hr />
</div>
<div id="edec" class="section level2">
<h2>5.2.18 <strong>Edec</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: EDec (Epigenomic Deconvolution) is a technique that, starting from methylation and gene expression profiles of bulk tissue samples, infers cell type composition of each input sample as well as DNA methylation and gene transcription profiles of constituent cell types.</p></li>
<li><p><strong>Installation</strong>: You can install the released version of Edec from GitHub with:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;BRL-BCM/EDec&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: EDec example application A example of an application of the EDec package can be found <a href="https://rawgit.com/BRL-BCM/EDec/master/doc/EDec_usage_example.html">here</a>.</li>
</ul>
<hr />
</div>
<div id="mmad" class="section level2">
<h2>5.2.19 <strong>MMAD</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: MMAD is capable of performing robust tissue micro-dissection in silico, and which can improve the detection of differentially expressed genes. MMAD software is implemented in MATLAB. The MMAD.zip file contains a brief user guide and MATLAB source code for implementation of MMAD.</p></li>
<li><p><strong>Installation</strong>: You can install the released version of MMAD from <a href="https://sourceforge.net/projects/mmad/">here</a>.</p></li>
</ul>
<hr />
</div>
<div id="scbio" class="section level2">
<h2>5.2.20 <strong>scBio</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Single Cell Genomics for Enhancing Cell Composition Inference from Bulk Genomics Data
Cellular population mapping (CPM) a deconvolution algorithm in which single-cell genomics is required in only one or a few samples, where in other samples of the same tissue, only bulk genomics is measured and the underlying fine resolution cellular heterogeneity is inferred.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;DSA&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://github.com/amitfrish/scBio">here</a>.</li>
</ul>
<hr />
</div>
<div id="debcam" class="section level2">
<h2>5.2.21 <strong>debCAM</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: An R package for fully unsupervised deconvolution of complex tissues. It provides basic functions to perform unsupervised deconvolution on mixture expression profiles by Convex Analysis of Mixtures (CAM) and some auxiliary functions to help understand the subpopulation-specific results. It also implements functions to perform supervised deconvolution based on prior knowledge of molecular markers, S matrix or A matrix. Combining molecular markers from CAM and from prior knowledge can achieve semi-supervised deconvolution of mixtures.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p></li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;debCAM&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/debCAM/inst/doc/debcam.html">here</a>.</li>
</ul>
<hr />
</div>
<div id="temt" class="section level2">
<h2>5.2.22 <strong>TEMT</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: TEMT (transcript Estimation from mixed tissue samples) is a probabilistic model-based approach, and estimate the transcript abundances of single cell type of interest via the RNA-Seq data from heterogeneous tissue sample. TEMT has incorporated the positional and sequence-specific bias, and with the online EM algorithm adopted, it has a time requirement proportional to the data size and a constant memory requirement.</p></li>
<li><p><strong>Installation</strong>:You can install the released version of TEMT from <a href="https://github.com/uci-cbcl/TEMT">here</a>.</p></li>
</ul>
<hr />
</div>
<div id="demixt" class="section level2">
<h2>5.2.23 <strong>DeMixT</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A deconvolution framework for mixed transcriptomes from heterogeneous tumor samples with two or three components using expression data from RNAseq or microarray platforms. DeMixT is a frequentist-based method and fast in yielding accurate estimates of cell proportions compart-ment-specific expression profiles for two-component and three-component deconvolution problem. Our method promises to provide deeper insight into cancer biomarkers and assist in the development of novel prognostic markers and therapeutic strategies. The function DeMixT is designed to finish the whole pipeline of deconvolution for two or three components. DeMixT.S1 function is designed to estimate the proportions of all mixed samples for each mixing component. DeMixT.S2 function is designed to estimate the component-specific deconvolved expressions of individual mixed samples for a given set of genes.</p></li>
<li><p><strong>Installation</strong>:DeMixT source files are compatible with windows, linux and mac os. For users who have OpenMP on the computer, please use <strong>DeMixT_0.2 (DeMixT_0.1 is an archived version)</strong>. To install this package, start R and enter:</p></li>
</ul>
<pre class="r"><code>devtools::install_github(&quot;wwylab/DeMixTallmaterials/DeMixT_0.2&quot;)</code></pre>
<p>For users who do not have OpenMP on the computer, please use <strong>DeMixT_0.2.1 (DeMixT_0.1.1 is an archived version)</strong>. To install this package, start R and enter:</p>
<pre class="r"><code>devtools::install_github(&quot;wwylab/DeMixTallmaterials/DeMixT_0.2.1&quot;)</code></pre>
<p>You can also download the installation files directly from the website:
DeMixT_0.2: <a href="http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.tar.gz" class="uri">http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.tar.gz</a>
DeMixT_0.2.1: <a href="http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.1.tar.gz" class="uri">http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.1.tar.gz</a></p>
<hr />
</div>
<div id="dtangle" class="section level2">
<h2>5.2.24 <strong>Dtangle</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Dtangle is a new deconvolution method that is accurate, robust and simple to compute. It estimates cell type proportions using biologically plausible models of high throughput profiling technology.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version 3.6 or higher) and enter:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;dtangle&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: <a href="dtangle.github.io">Vignettes</a>:
+Bulk RNA-seq and microarrays: <a href="https://gjhunt.github.io/dtangle/vign/basic-deconvolution.html">here</a>
+Using single-cell RNA-seq as references: <a href="https://gjhunt.github.io/dtangle/vign/sc_vignette.html">here</a>
+PBMC microarray deconvolution: <a href="https://gjhunt.github.io/dtangle/vign/blood.html">here</a></li>
</ul>
<hr />
</div>
<div id="fardeep" class="section level2">
<h2>5.2.25 <strong>FARDEEP</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: FARDEEP (Fast And Robust DEconvolution of Expression Profiles) is a new machine learning tool for enumerating immune cell subsets from whole tumor tissue samples. FARDEEP utilizes an adaptive least trimmed square to automatically detect and remove outliers before estimating the cell compositions.</p></li>
<li><p><strong>Installation</strong>: The source code for FARDEEP is implemented in “R” and available for download at <a href="https://github.com/YuningHao/FARDEEP.git" class="uri">https://github.com/YuningHao/FARDEEP.git</a>.</p></li>
</ul>
<hr />
</div>
<div id="toast" class="section level2">
<h2>5.2.26 <strong>TOAST</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: This package is devoted to analyzing high-throughput data (e.g. gene expression microarray, DNA methylation microarray, RNA-seq) from complex tissues. Current functionalities include 1. detect cell-type specific or cross-cell type differential signals 2. tree-based differential analysis 3. improve variable selection in reference-free deconvolution 4. partial reference-free deconvolution with prior knowledge.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version “4.2”) and enter:</p></li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;TOAST&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/TOAST/inst/doc/TOAST.html">here</a>.</li>
</ul>
<hr />
</div>
<div id="cdseq" class="section level2">
<h2>5.2.27 <strong>CDSeq</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: CDSeq is a complete deconvolution method for dissecting bulk RNA-Seq data. The input of CDSeq is, ideally, bulk RNA-Seq read counts (similar to the input format required by DESeq2), and CDSeq will estimate, simultaneously, the cell-type-specific gene expression profiles and the sample-specific cell-type proportions, no reference of pure cell line GEPs or scRNAseq reference is needed for running CDSeq.</p></li>
<li><p><strong>Installation</strong>: You can install the released version of CDSeq from CRAN with:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;CDSeq&quot;)</code></pre>
<p>And the development version from GitHub with:</p>
<pre class="r"><code># install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;kkang7/CDSeq_R_Package&quot;)</code></pre>
<p>build the vignette with</p>
<pre class="r"><code># install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;kkang7/CDSeq_R_Package&quot;, build_vignettes = TRUE)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://github.com/kkang7/CDSeq_R_Package">here</a>.</li>
</ul>
<hr />
</div>
<div id="casey-p.-shannon-et-al." class="section level2">
<h2>5.2.28 <strong>Casey P. Shannon <em>et al.</em> </strong></h2>
<ul>
<li><strong>Introduction</strong>: A two-stage, in silico deconvolution approach that first predicts sample composition to biologically meaningful and homogeneous leukocyte sub-populations, and then performs cell type-specific differential expression analysis in these same sub-populations, from peripheral whole blood expression data.</li>
<li><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://doi.org/10.1371/journal.pone.0095224" class="uri">https://doi.org/10.1371/journal.pone.0095224</a></li>
</ul>
<hr />
</div>
<div id="deconvseq" class="section level2">
<h2>5.2.29 <strong>deconvSeq</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: R package for performing cell type deconvolution of bulk RNA sequencing, single cell RNA sequencing, and bisulfite sequencing data.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version “4.2”) and enter:</p></li>
</ul>
<pre class="r"><code>#in R
library(devtools)
install_deps(&quot;/PATH/TO/deconvSeq&quot;,dependencies=TRUE)
#on command line
#R CMD INSTALL deconvSeq_VERSION_NUMBER.tar.gz</code></pre>
<p>Alternatively, use the devtools package to install from GitHub:</p>
<pre class="r"><code>## devtools is required
library(devtools)
install_github(&quot;rosedu1/deconvSeq&quot;, dependencies=TRUE)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://rosedu1.github.io/deconvSeq_vignette.html">here</a>.</li>
</ul>
<hr />
</div>
<div id="xiaoqing-yu-et-al." class="section level2">
<h2>5.2.30 <strong>Xiaoqing Yu <em>et al.</em> </strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Xiaoqing Yu <em>et al.</em> introduce a scheme for characterizing cell compositions from bulk tumor gene expression by integrating signatures learned from scRNA-seq data.</p></li>
<li><p><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://doi.org/10.1186/s12885-019-5927-3" class="uri">https://doi.org/10.1186/s12885-019-5927-3</a></p></li>
</ul>
<hr />
</div>
<div id="digitaldlsorter" class="section level2">
<h2>5.2.31 <strong>Digitaldlsorter</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A pipeline to generate a Deep Neural Network cell type deconvolution model for bulk RNASeq samples from single cell rna-seq data.</p></li>
<li><p><strong>Installation</strong>: An implementation of the Digitaldlsorter in R can be found here: <a href="https://github.com/cartof/digitalDLSorter" class="uri">https://github.com/cartof/digitalDLSorter</a></p></li>
<li><p><strong>Application</strong>: More examples can download from <a href="https://github.com/cartof/digitalDLSorter">here</a>.</p></li>
</ul>
<hr />
</div>
<div id="declust" class="section level2">
<h2>5.2.32 <strong>DeClust</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A reference-free deconvolution method to infer cancer cell-intrinsic subtypes and tumor-type-specific stromal profiles.</p></li>
<li><p><strong>Installation</strong>: An implementation of the DeClust can be found here: <a href="https://github.com/integrativenetworkbiology/DeClust" class="uri">https://github.com/integrativenetworkbiology/DeClust</a></p></li>
</ul>
<hr />
</div>
<div id="mysort" class="section level2">
<h2>5.2.33 <strong>MySort</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: MySort is a R program to estimate the fraction and abundance of immune cells based on gene expression profile in heterogeneous mixture tissues.</p></li>
<li><p><strong>Installation</strong>: Clone this repository: “hg clone <a href="https://testtoolshed.g2.bx.psu.edu/repos/moneycat/mysort" class="uri">https://testtoolshed.g2.bx.psu.edu/repos/moneycat/mysort</a>”.</p></li>
</ul>
<hr />
</div>
<div id="smcgenedeconv" class="section level2">
<h2>5.2.34 <strong>smcgenedeconv</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A sequential Monte Carlo approach to gene expression deconvolution.</p></li>
<li><p><strong>Installation</strong>: An implementation of the smcgenedeconv in R can be found here: <a href="https://github.com/moyanre/smcgenedeconv" class="uri">https://github.com/moyanre/smcgenedeconv</a></p></li>
</ul>
<hr />
</div>
<div id="chiu-yen-jung-et-al." class="section level2">
<h2>5.2.35 <strong>Chiu Yen-Jung <em>et al.</em> </strong></h2>
<ul>
<li><p><strong>Introduction</strong>: The tool is designed to estimate the fractions of immune cells from the expression profiles of tissues, such as tumor masses, PBMCs, etc. The 8 immune cells treated in this method include naïve CD8 T cells, naïve CD4 T cells, alternatively activated macrophages, classically activated macrophages, regulatory T cells, T helper cells, natural killer cells, and dendritic cells.</p></li>
<li><p><strong>Installation</strong>: An implementation of in python and R can be found here: <a href="https://github.com/holiday01/deconvolution-to-estimate-immune-cell-subsets" class="uri">https://github.com/holiday01/deconvolution-to-estimate-immune-cell-subsets</a></p></li>
</ul>
<hr />
</div>
<div id="nitumid" class="section level2">
<h2>5.2.36 <strong>NITUMID</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: NITUMID is the R implementation of Non-negative Matrix Factorization-based Immune-TUmor MIcroenvironment Deconvolution, a statistical framework for tumor immune microenvironment deconvolution. NITUMID takes gene expression matrix (either from RNA-Seq or microarray) as input, and use our own curated list of signature genes to estimate proportions of immune and tumor cells, as well as their corresponding mRNA levels. Current version of NITUMID only supports tumor microenvironment deconvolution for bulk melanoma data. And current framework includes 11 component cell types: Dendritic cells, CD4+ T cell, CD8+ T cell, Macrophages, B cells, Natural killer cells (NK/NKT), Monocytes, Plasma, MAST cell, Eosinophils and melanoma.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version “4.2”) and enter:</p></li>
</ul>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
library(&quot;devtools&quot;)

install_github(&quot;tdw1221/NITUMID&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: For detailed introduction and tutorial, please see NITUMID’s vignette at <a href="https://github.com/tdw1221/NITUMID/blob/master/docs/nitumid-tutorial.html" class="uri">https://github.com/tdw1221/NITUMID/blob/master/docs/nitumid-tutorial.html</a></li>
</ul>
<hr />
</div>
<div id="adapts" class="section level2">
<h2>5.2.37 <strong>ADAPTS</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Augments existing or de-novo cell-type signature matrices to deconvolve bulk gene expression data This package expands on the techniques outlined in Newman et al.’s 2015 Nature Methods paper: ‘Robust enumeration of cell subsets from tissue expression profiles’. to allow a user to easily add their own cell types (e.g. a tumor specific cell type) to Newman’s LM22 or other signature matrix.</p></li>
<li><p><strong>Installation</strong>: To install this package, start R (version “4.2”) and enter:</p></li>
</ul>
<pre class="r"><code>installed.packages(ADAPTS)</code></pre>
<p>To install this package in R, use devtools.</p>
<pre class="r"><code>install.packages(&#39;devtools&#39;)

library(devtools)

devtools::install_github(&#39;sdanzige/ADAPTS&#39;)</code></pre>
<hr />
</div>
<div id="momf" class="section level2">
<h2>5.2.38 <strong>MOMF</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: MOMF (Multi-Omics Matrix Factorization) is a package, to estimate the cell-type compositions of bulk RNA sequencing (RNA-seq) data by leveraging cell type-specific gene expression levels from single-cell RNA sequencing (scRNA-seq) data. MOMF not only directly models the count nature of gene expression data, but also effectively accounts for the uncertainty of cell type-specific mean gene expression levels.</p></li>
<li><p><strong>Installation</strong>:</p></li>
</ul>
<pre class="r"><code>### install devtools packages (devtools package)
install.packages(&quot;devtools&quot;)

### install MOMF package
devtools::install_github(&quot;sqsun/MOMF&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://github.com/sqsun/MOMF" class="uri">https://github.com/sqsun/MOMF</a>.</li>
</ul>
<hr />
</div>
<div id="deblender" class="section level2">
<h2>5.2.39 <strong>Deblender</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Deblender is a flexible complete deconvolution tool operating in semi−/unsupervised mode based on the user’s access to known marker gene lists and information about cell/tissue composition. In case of no prior knowledge, global gene expression variability is used in clustering the mixed data to substitute marker sets with cluster sets.</p></li>
<li><p><strong>Installation</strong>: Deblender is implemented in MATLAB and is available from <a href="https://github.com/kondim1983/Deblender/" class="uri">https://github.com/kondim1983/Deblender/</a>.</p></li>
</ul>
<hr />
</div>
<div id="lishuang-qi-et-al." class="section level2">
<h2>5.2.40 <strong>Lishuang Qi <em>et al.</em> </strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A modified semi-supervised nonnegative matrix factorization method for gene expression deconvolution.</p></li>
<li><p><strong>Installation</strong>: The deconvolution algorithms can be found here: <a href="https://doi.org/10.1371/journal.pone.0100934" class="uri">https://doi.org/10.1371/journal.pone.0100934</a></p></li>
</ul>
<hr />
</div>
<div id="mixture" class="section level2">
<h2>5.2.41 <strong>MIXTURE</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: An analytical method based on a noise constrained recursive variable selection for a support vector regression. The MIXTURE shiny App has been only tested on Linux. The RUN_MIXTURE code was tested on Linux, Windows and Mac. On windows only one CPU core is allowed.</p></li>
<li><p><strong>Installation</strong>: MIXTURE is implemented in Python and R at <a href="https://github.com/elmerfer/MIXTURE.App" class="uri">https://github.com/elmerfer/MIXTURE.App</a>.</p></li>
<li><p><strong>Application</strong>: More examples can download from <a href="https://github.com/elmerfer/MIXTURE.App" class="uri">https://github.com/elmerfer/MIXTURE.App</a>.</p></li>
</ul>
<hr />
</div>
<div id="semi-cam" class="section level2">
<h2>5.2.42 <strong>Semi-CAM</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A semi-supervised deconvolution method for bulk transcriptomic data with partial marker gene information.</p></li>
<li><p><strong>Installation</strong>: Source R code and download the required package:</p></li>
</ul>
<pre class="r"><code>library(devtools)
#or install.packages(&quot;devtools&quot;)
R_githubURL &lt;- &quot;https://raw.github.com/ylidong/semi-CAM/master/FUNS_new/&quot;

source_url(paste0(R_githubURL,&quot;proportion_estimate_main.R&quot;))
source_url(paste0(R_githubURL,&quot;measure_conv_miss.R&quot;))
source_url(paste0(R_githubURL,&quot;identify_markers.R&quot;))
source_url(paste0(R_githubURL,&quot;small.fun.R&quot;))
source_url(paste0(R_githubURL,&quot;match_celltype.R&quot;))

#########################################
##download the required CellMix package##
#########################################
# get biocLite from bioconductor
source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;BiocInstaller&quot;)
# or
library(BiocInstaller)

# install the required package CellMix
# On Windows or Mac OS X, this will try install the source package 
biocLite(&#39;CellMix&#39;, siteRepos=c(&#39;http://web.cbio.uct.ac.za/~renaud/CRAN&#39;), type=&#39;both&#39;)

# on Linux this will try install the source package 
biocLite(&#39;CellMix&#39;, siteRepos=c(&#39;http://web.cbio.uct.ac.za/~renaud/CRAN&#39;))

# if cannot open the link, can download the source codes from the folder Required_pacakages_source and install it using 
#install.packages(&quot;path_to_file/NMF_0.20.tar.gz&quot;, repos = NULL, type=&quot;source&quot;)
#install.packages(&quot;path_to_file/CellMix_1.6.2.tar.gz&quot;, repos = NULL, type=&quot;source&quot;)

library(CellMix)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://github.com/ylidong/semi-CAM" class="uri">https://github.com/ylidong/semi-CAM</a>.</li>
</ul>
<hr />
</div>
<div id="dtd" class="section level2">
<h2>5.2.43 <strong>DTD</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: DTD (Digital Tissue Deconvolution) reconstructs the cellular composition of a tissue from its bulk expression profile. In order to increase deconvolution accuracy, DTD adapts the deconvolution model to the tissue scenario via loss-function learning.
Training is performed on ‘in-silicio’ training mixtures, for which the cellular composition are known.
As input, DTD requires a labelled expression matrix. The package includes functions to generate training and test mixtures, train the model, and assess its deconvolution capability via visualizations.</p></li>
<li><p><strong>Installation</strong>: Install from github, without vignette:</p></li>
</ul>
<pre class="r"><code>devtools::install_github(&quot;spang-lab/DTD&quot;)</code></pre>
<p>Install from github with vignette (creating vignettes approximately takes ~3 minutes):</p>
<pre class="r"><code>devtools::install_github(
    &quot;spang-lab/DTD&quot;, 
    build_opts = c(&quot;--no-resave-data&quot;, &quot;--no-manual&quot;), 
    build_vignettes=TRUE)

browseVignettes(&quot;DTD&quot;)</code></pre>
<ul>
<li><strong>Application</strong>: More examples can download from <a href="https://htmlpreview.github.io/?https://github.com/MarianSchoen/Exemplary-DTD-analysis/blob/master/2019_04_12_DTD_melanoma.html" class="uri">https://htmlpreview.github.io/?https://github.com/MarianSchoen/Exemplary-DTD-analysis/blob/master/2019_04_12_DTD_melanoma.html</a>.</li>
</ul>
<hr />
</div>
<div id="infino" class="section level2">
<h2>5.2.44 <strong>Infino</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: DTD (Digital Tissue Deconvolution) reconstructs the cellular composition of a tissue from its bulk expression profile. In order to increase deconvolution accuracy, DTD adapts the deconvolution model to the tissue scenario via loss-function learning.
Training is performed on ‘in-silicio’ training mixtures, for which the cellular composition are known.
As input, DTD requires a labelled expression matrix. The package includes functions to generate training and test mixtures, train the model, and assess its deconvolution capability via visualizations.</p></li>
<li><p><strong>Installation</strong>: A Docker image at: <a href="https://hub.docker.com/r/hammerlab/infino-docker/" class="uri">https://hub.docker.com/r/hammerlab/infino-docker/</a>.</p></li>
<li><p><strong>Application</strong>: More examples can download from <a href="https://github.com/hammerlab/infino" class="uri">https://github.com/hammerlab/infino</a>.</p></li>
</ul>
</div>


<footer class="footline">
	
</footer>

        
        </div>
        

      </div>

    <div id="navigation">
        
        

        
            
            
                
                    
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                        
                        
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                        
                        
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                        
                        
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        

        


	 
	 
		
			<a class="nav nav-prev" href="../../conditional-analysis/spatialandtemporal/" title="Spatial and temporal variation"> <i class="fa fa-chevron-left"></i></a>
		
		
			<a class="nav nav-next" href="../../conditional-analysis/trajectory/" title="Tajectory inference" style="margin-right: 0px;"><i class="fa fa-chevron-right"></i></a>
		
	
    </div>

    </section>

    <div style="left: -1000px; overflow: scroll; position: absolute; top: -1000px; border: none; box-sizing: content-box; height: 200px; margin: 0px; padding: 0px; width: 200px;">
      <div style="border: none; box-sizing: content-box; height: 200px; margin: 0px; padding: 0px; width: 200px;"></div>
    </div>
    <script src="../../js/clipboard.min.js?1672904510"></script>
    <script src="../../js/perfect-scrollbar.min.js?1672904510"></script>
    <script src="../../js/perfect-scrollbar.jquery.min.js?1672904510"></script>
    <script src="../../js/jquery.sticky.js?1672904510"></script>
    <script src="../../js/featherlight.min.js?1672904510"></script>
    <script src="../../js/highlight.pack.js?1672904510"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script src="../../js/modernizr.custom-3.6.0.js?1672904510"></script>
    <script src="../../js/learn.js?1672904510"></script>
    <script src="../../js/hugo-learn.js?1672904510"></script>
    
        
            <script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js"></script>
        
        <script>
            mermaid.initialize({ startOnLoad: true });
        </script>
    
    

  </body>
</html>

