---
title: "ceRNA-gene interaction database"
output:
  blogdown::html_page:
    toc: true
weight: 4
---


<div id="TOC">
<ul>
<li><a href="#encode" id="toc-encode">2.4.1 <strong>ENCODE</strong></a></li>
<li><a href="#starbase" id="toc-starbase">2.4.2 <strong>starBase</strong></a></li>
<li><a href="#mircode" id="toc-mircode">2.4.3 <strong>miRcode</strong></a></li>
<li><a href="#diana-lncbase" id="toc-diana-lncbase">2.4.4 <strong>DIANA-LncBase</strong></a></li>
<li><a href="#omnipath" id="toc-omnipath">2.4.5 <strong>OmniPath</strong></a></li>
<li><a href="#cerdb" id="toc-cerdb">2.4.6 <strong>ceRDB</strong></a></li>
<li><a href="#lncactdb" id="toc-lncactdb">2.4.7 <strong>LncACTdb</strong></a></li>
<li><a href="#pcerbase" id="toc-pcerbase">2.4.8 <strong>PceRBase</strong></a></li>
</ul>
</div>

<hr />
<div id="encode" class="section level2">
<h2>2.4.1 <strong>ENCODE</strong></h2>
<p><strong>Introduction</strong>: The ENCODE Portal, developed and maintained by the Data Coordination Center (ENCODE DCC), is the canonical source for all experimental metadata and data from ENCODE and associated projects. The ENCODE Portal contains raw and ground-level analysis data generated by participating mapping centers using a wide-range of assays (integrative analysis data are available at the SCREEN portal. The Portal also stores records of the materials and methods used to perform the assays and subsequent analysis.<br />
<img src="/images/ENCODE.jpg?width=50pc" alt="ENCODE" /></p>
<p><strong>Application</strong>: The ENCODE tutorial can be found at <a href="https://www.encodeproject.org/help/getting-started/" class="uri">https://www.encodeproject.org/help/getting-started/</a>.</p>
<p><strong>Web server</strong>: <a href="https://www.encodeproject.org/">ENCODE</a>.</p>
<hr />
</div>
<div id="starbase" class="section level2">
<h2>2.4.2 <strong>starBase</strong></h2>
<p><strong>Introduction</strong>: starBase is an openly licensed and state-of-the-art platform to decode the cellular interaction networks among thousands of RNA binding proteins (RBPs) and various RNAs. We have identified millions of RBP-RNA and RNA-RNA interactions through analyzing thousands of CLIP-seq and various high-throughput sequencing data by developing new software and tools, and studied their functions and mechanisms in human diseases.</p>
<center>
<b>Database information</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Species</td>
<td align="left">23</td>
</tr>
<tr class="even">
<td align="left">CLIP-seq data</td>
<td align="left">&gt;700 datasets</td>
</tr>
<tr class="odd">
<td align="left">Degradome-seq</td>
<td align="left">100 datasets</td>
</tr>
<tr class="even">
<td align="left">RNA-RNA interactome</td>
<td align="left">&gt;30 datasets</td>
</tr>
<tr class="odd">
<td align="left">RNA-seq data</td>
<td align="left">&gt;10,800 samples from 32 cancer types</td>
</tr>
<tr class="even">
<td align="left">miRNA-seq data</td>
<td align="left">&gt;10,500 samples from 32 cancer types</td>
</tr>
<tr class="odd">
<td align="left">Disease data</td>
<td align="left">&gt;3,236,000 mutations from 366 disease types</td>
</tr>
<tr class="even">
<td align="left">miRNA-ncRNA(CLIP)</td>
<td align="left">&gt;1,100,000 interactions</td>
</tr>
<tr class="odd">
<td align="left">miRNA-mRNA(CLIP)</td>
<td align="left">&gt;2,500,000 interactions</td>
</tr>
<tr class="even">
<td align="left">RBP-mRNA</td>
<td align="left">&gt;1,208,900 interactions</td>
</tr>
<tr class="odd">
<td align="left">RBP-ncRNA</td>
<td align="left">&gt;117,000 interactions</td>
</tr>
<tr class="even">
<td align="left">RNA-RNA</td>
<td align="left">&gt;1,530,000 interactions</td>
</tr>
</tbody>
</table>
<p><strong>Application</strong>: The starBase tutorial can be found at <a href="https://starbase.sysu.edu.cn/tutorial.php" class="uri">https://starbase.sysu.edu.cn/tutorial.php</a>.</p>
<p><strong>Web server</strong>: <a href="https://starbase.sysu.edu.cn/index.php">starBase</a>.
<img src="/images/starBase.jpg?width=50pc" alt="starBase" /></p>
<hr />
</div>
<div id="mircode" class="section level2">
<h2>2.4.3 <strong>miRcode</strong></h2>
<p><strong>Introduction</strong>: MiRcode provides “whole transcriptome” human microRNA target predictions based on the comprehensive GENCODE gene annotation, including &gt;10,000 long non-coding RNA genes. Coding genes are also covered, including atypical regions such as 5’UTRs and CDS. MicroRNA family definitions and names are consistent with TargetScan. Site conservation is evaluated based on 46 vertebrates species. Data sources and stats is found on the data info page, and the FAQ answers many questions. Please cite the miRcode paper if this resource is useful in your research.</p>
<center>
<b>Database information</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Species</td>
<td align="left">10</td>
</tr>
<tr class="even">
<td align="left">Total transcripts</td>
<td align="left">179,905</td>
</tr>
<tr class="odd">
<td align="left">Total genes</td>
<td align="left">53,520</td>
</tr>
<tr class="even">
<td align="left">LncRNA genes</td>
<td align="left">10,419</td>
</tr>
<tr class="odd">
<td align="left">Coding genes</td>
<td align="left">19,999</td>
</tr>
<tr class="even">
<td align="left">Pseudogenes</td>
<td align="left">12,549</td>
</tr>
</tbody>
</table>
<p><strong>Web server</strong>: <a href="http://www.mircode.org/">miRcode</a>.
<img src="/images/miRcode.jpg?width=50pc" alt="miRcode" /></p>
<hr />
</div>
<div id="diana-lncbase" class="section level2">
<h2>2.4.4 <strong>DIANA-LncBase</strong></h2>
<p><strong>Introduction</strong>: DIANA-LncBase is a reference repository with experimentally supported miRNA targets on non-coding transcripts. It catalogues approximately ~500,000 entries, corresponding to ~240,000 unique tissue and cell-type specific miRNA-lncRNA interactions. The incorporated interactions are defined by 15 distinct low-/high-throughput methodologies, corresponding to 243 distinct cell types/tissues and 162 experimental conditions.</p>
<center>
<b>Database information</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">miRNA-lncRNA entries</td>
<td align="left">730</td>
</tr>
<tr class="even">
<td align="left">miRNA-lncRNA chimeric fragments</td>
<td align="left">2924</td>
</tr>
</tbody>
</table>
<p><strong>Application</strong>: The DIANA-LncBase tutorial can be found at
<a href="https://diana.e-ce.uth.gr/lncbasev3/help" class="uri">https://diana.e-ce.uth.gr/lncbasev3/help</a>.</p>
<p><strong>Web server</strong>: <a href="https://diana.e-ce.uth.gr/lncbasev3">DIANA-LncBase</a>.
<img src="/images/DIANA-LncBase.jpg?width=50pc" alt="DIANA-LncBase" /></p>
<hr />
</div>
<div id="omnipath" class="section level2">
<h2>2.4.5 <strong>OmniPath</strong></h2>
<p><strong>Introduction</strong>: OmniPath is a database of molecular biology prior knowledge. It combines data from over 100 resources and builds 5 integrated databases: signaling network (interactions), enzyme-PTM relationships, protein complexes, protein annotations (function, localization, tissue, disease, structure, etc), and intercellular communication roles (e.g. ligand, receptor; intercell).<br />
<img src="/images/OmniPath1.jpg?width=50pc" alt="OmniPath1" /></p>
<center>
<b>Database information</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Transcriptional regulation</td>
<td align="left">8</td>
</tr>
<tr class="even">
<td align="left">RNA regulation</td>
<td align="left">6</td>
</tr>
<tr class="odd">
<td align="left">Ligand-receptor</td>
<td align="left">6</td>
</tr>
<tr class="even">
<td align="left">Protein-protein interaction resources</td>
<td align="left">44</td>
</tr>
<tr class="odd">
<td align="left">Extracellular</td>
<td align="left">14</td>
</tr>
</tbody>
</table>
<p><strong>Installation</strong>: 1.Start R (version “4.2”) and enter:</p>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;OmnipathR&quot;)</code></pre>
<p>Or install R package from <a href="https://github.com/saezlab/OmnipathR">github</a>:</p>
<pre class="r"><code>require(devtools)
install_github(&#39;saezlab/OmnipathR&#39;)</code></pre>
<p>2.Python client for the OmniPath web service:</p>
<pre class="python"><code>pip install omnipath</code></pre>
<p>Or install Python package from <a href="https://github.com/saezlab/omnipath">github</a>.
3. A Cytoscape plug-in <a href="https://apps.cytoscape.org/apps/omnipath">here</a>.</p>
<p><strong>Application</strong>: The OmniPath tutorial can be found at
<a href="https://workflows.omnipathdb.org/" class="uri">https://workflows.omnipathdb.org/</a>.
OmnipathR vignette can be found at <a href="http://bioconductor.org/packages/release/bioc/vignettes/OmnipathR/inst/doc/nichenet.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/OmnipathR/inst/doc/nichenet.html</a>, <a href="http://bioconductor.org/packages/release/bioc/vignettes/OmnipathR/inst/doc/db_manager.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/OmnipathR/inst/doc/db_manager.html</a> and <a href="http://bioconductor.org/packages/release/bioc/vignettes/OmnipathR/inst/doc/drug_targets.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/OmnipathR/inst/doc/drug_targets.html</a>.</p>
<p><strong>Web server</strong>: <a href="https://omnipathdb.org/">OmniPath</a>.
<img src="/images/OmniPath.jpg?width=60pc" alt="OmniPath" /></p>
<hr />
</div>
<div id="cerdb" class="section level2">
<h2>2.4.6 <strong>ceRDB</strong></h2>
<p><strong>Introduction</strong>: By examining the co-occurrence of miRNA response elements in the mRNAs on a genome-wide basis we predict competing endogenous RNA for specific mRNAs targeted by miRNAs.</p>
<p><strong>Web server</strong>: <a href="https://www.oncomir.umn.edu/cefinder/basic_search.php">ceRDB</a>.</p>
<hr />
</div>
<div id="lncactdb" class="section level2">
<h2>2.4.7 <strong>LncACTdb</strong></h2>
<p><strong>Introduction</strong>: LncACTdb 3.0 is a comprehensive database of experimentally supported competing endogenous RNA (ceRNA) interactions and personalized networks contributing to precision medicine. LncACTdb is a continually updated database that is used specifically to store experimentally validated ceRNA interactions manually curated from different species and disease states.
<img src="/images/LncACTdb1.jpg?width=40pc" alt="LncACTdb1" />
<img src="/images/LncACTdb2.png?width=40pc" alt="LncACTdb2" /></p>
<p><strong>Application</strong>: The LncACTdb tutorial can be found at <a href="https://starbase.sysu.edu.cn/tutorial.php" class="uri">https://starbase.sysu.edu.cn/tutorial.php</a>.</p>
<p><strong>Web server</strong>: <a href="http://bio-bigdata.hrbmu.edu.cn/LncACTdb/">LncACTdb</a>.
<img src="/images/LncACTdb.jpg?width=40pc" alt="LncACTdb" /></p>
<hr />
</div>
<div id="pcerbase" class="section level2">
<h2>2.4.8 <strong>PceRBase</strong></h2>
<p><strong>Introduction</strong>: Competing endogenous RNA (ceRNA) act as decoys for microRNA (miRNA) binding, thereby regulating the abundance of other RNA transcripts which share the same microRNA response elements (MREs). Plant ceRNA database (PceRBase) uses mRNA, long non-coding RNA and miRNA sequence information to predict potential ceRNA target-target pairs (with identical shared MREs) and ceRNA target-mimic pairs (with one perfect MRE match, and one imperfect MRE match).</p>
<p>PceRBase currently predicts 696,756 potential ceRNA target-target, and 167,608 ceRNA target-mimic pairs from 28 plant species. Particularly, 239,998 ceRNA target-target and 75,360 ceRNA target-mimic pairs were found in Arabidopsis thaliana.
<img src="/images/PceRBase.jpg?width=50pc" alt="PceRBase" /></p>
<p><strong>Application</strong>: The PceRBase tutorial can be found at <a href="http://bis.zju.edu.cn/pcernadb/document/index.jsp" class="uri">http://bis.zju.edu.cn/pcernadb/document/index.jsp</a>.</p>
<p><strong>Web server</strong>: <a href="http://bis.zju.edu.cn/pcernadb/index.jsp">PceRBase</a>.
<img src="/images/PceRBase1.jpg?width=50pc" alt="PceRBase1" /></p>
</div>
