---
title: "miRNA-gene interaction database"
output:
  blogdown::html_page:
    toc: true
weight: 3
---


<div id="TOC">
<ul>
<li><a href="#targetscan" id="toc-targetscan">3.3.1 <strong>TargetScan</strong></a></li>
<li><a href="#mirbase" id="toc-mirbase">3.3.2 <strong>miRBase</strong></a></li>
<li><a href="#mirdb" id="toc-mirdb">3.3.3 <strong>miRDB</strong></a></li>
<li><a href="#rnahybrid" id="toc-rnahybrid">3.3.4 <strong>RNAhybrid</strong></a></li>
<li><a href="#mirtarbase" id="toc-mirtarbase">3.3.5 <strong>miRTarbase</strong></a></li>
<li><a href="#mir2disease" id="toc-mir2disease">3.3.6 <strong>miR2Disease</strong></a></li>
<li><a href="#tarbase" id="toc-tarbase">3.3.7 <strong>TarBase</strong></a></li>
<li><a href="#mirdip" id="toc-mirdip">3.3.8 <strong>mirDIP</strong></a></li>
</ul>
</div>

<p><strong>Breaking</strong>:</p>
<hr />
<div id="targetscan" class="section level2">
<h2>3.3.1 <strong>TargetScan</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: TargetScan predicts biological targets of miRNAs by searching for the presence of conserved 8mer, 7mer, and 6mer sites that match the seed region of each miRNA (Lewis et al., 2005). As options, predictions with only poorly conserved sites and predictions with nonconserved miRNAs are also provided. Also identified are sites with mismatches in the seed region that are compensated by conserved 3’ pairing (Friedman et al., 2009).</p></li>
<li><p><strong>Installation</strong>: A GitHub package to visualize mRNA fold changes in response to a miRNA perturbation, compute Pct parameters, train regression models to predict miRNA targets, and compare the relative performances of miRNA target prediction methods at <a href="https://github.com/vagarwal87/TargetScanTools" class="uri">https://github.com/vagarwal87/TargetScanTools</a>.</p></li>
<li><p><strong>Application</strong>: The TargetScan tutorial can be found at <a href="https://www.targetscan.org/faqs.Release_8.html" class="uri">https://www.targetscan.org/faqs.Release_8.html</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="https://www.targetscan.org/vert_80/">TargetScan</a>.
<img src="/images/TargetScan.jpg?width=50pc" alt="TargetScan" /></p></li>
</ul>
<hr />
</div>
<div id="mirbase" class="section level2">
<h2>3.3.2 <strong>miRBase</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: The miRBase database aims to provide integrated interfaces to comprehensive microRNA sequence data, annotation and predicted gene targets. miRBase takes over functionality from the microRNA Registry and fulfils three main roles: the miRBase Registry acts as an independent arbiter of microRNA gene nomenclature, assigning names prior to publication of novel miRNA sequences. miRBase Sequences is the primary online repository for miRNA sequence data and annotation. miRBase Targets is a comprehensive new database of predicted miRNA target genes. miRBase catalogs, names and distributes microRNA gene sequences. The latest release of miRBase (v22) contains microRNA sequences from 271 organisms: 38 589 hairpin precursors and 48 860 mature microRNAs.</p></li>
<li><p><strong>Application</strong>: The miRBase tutorial can be found at
<a href="https://www.mirbase.org/help/index.shtml" class="uri">https://www.mirbase.org/help/index.shtml</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="https://www.mirbase.org/">miRBase</a>.
<img src="/images/TarBase1.jpg?width=100pc" alt="miRBase" /></p></li>
</ul>
<hr />
</div>
<div id="mirdb" class="section level2">
<h2>3.3.3 <strong>miRDB</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: miRDB is an online database for miRNA target prediction and functional annotations. All the targets in miRDB were predicted by a bioinformatics tool, MirTarget, which was developed by analyzing thousands of miRNA-target interactions from high-throughput sequencing experiments. Common features associated with miRNA binding and target downregulation have been identified and used to predict miRNA targets with machine learning methods. miRDB hosts predicted miRNA targets in five species: human, mouse, rat, dog and chicken. Users may also provide their own sequences for custom target prediction using the updated prediction algorithm. In addition, through combined computational analyses and literature mining, functionally active miRNAs in humans and mice were identified. These miRNAs, as well as associated functional annotations, are presented in the FuncMir Collection in miRDB.</p>
<center>
<p><b>Database information</b></p>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Species</td>
<td align="left">5</td>
</tr>
<tr class="even">
<td align="left">MicroRNA With Target</td>
<td align="left">7086</td>
</tr>
<tr class="odd">
<td align="left">Gene Target</td>
<td align="left">3519884</td>
</tr>
<tr class="even">
<td align="left">Unique Gene Target</td>
<td align="left">102559</td>
</tr>
</tbody>
</table></li>
<li><p><strong>Application</strong>: The miRDB tutorial can be found at
<a href="https://mirdb.org/faq.html#How_to_perform_target_search" class="uri">https://mirdb.org/faq.html#How_to_perform_target_search</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="http://www.mirdb.org/">miRDB</a>.
<img src="/images/miRDB.jpg?width=90pc" alt="miRDB" /></p></li>
</ul>
<hr />
</div>
<div id="rnahybrid" class="section level2">
<h2>3.3.4 <strong>RNAhybrid</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: RNAhybrid is a tool for finding minimum free energy hybridization of a long (target) and a short (query) RNA. The hybridization is performed in a kind of domain mode, ie. the short sequence is hybridized to the best fitting parts of the long one. The tool is primarily meant as a means for microRNA target prediction.
<img src="/images/miRTarbase.jpg?width=100pc" alt="miRTarbase" /></p></li>
<li><p><strong>Installation</strong>: <a href="https://bibiserv.cebitec.uni-bielefeld.de/applications/rnahybrid/resources/downloads/RNAhybrid-2.1.2.tar.gz">RNAhybrid src package</a> should be compiled on almost every unix like system (eg. Linux, OSX, Win+CygWin)</p></li>
<li><p><strong>Application</strong>: The RNAhybrid tutorial can be found at
<a href="https://bibiserv.cebitec.uni-bielefeld.de/rnahybrid?id=rnahybrid_manual_manual" class="uri">https://bibiserv.cebitec.uni-bielefeld.de/rnahybrid?id=rnahybrid_manual_manual</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="https://bibiserv.cebitec.uni-bielefeld.de/rnahybrid">RNAhybrid</a>.</p></li>
</ul>
<hr />
</div>
<div id="mirtarbase" class="section level2">
<h2>3.3.5 <strong>miRTarbase</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: As a database, miRTarBase has accumulated more than three hundred and sixty thousand miRNA-target interactions (MTIs), which are collected by manually surveying pertinent literature after NLP of the text systematically to filter research articles related to functional studies of miRNAs. Generally, the collected MTIs are validated experimentally by reporter assay, western blot, microarray and next-generation sequencing experiments. While containing the largest amount of validated MTIs, the miRTarBase provides the most updated collection by comparing with other similar, previously developed databases.<br />
<img src="/images/miRTarbase.jpg?width=100pc" alt="miRTarbase" /></p></li>
<li><p><strong>Application</strong>: The miRTarbase tutorial can be found at
<a href="https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2022/php/help.php" class="uri">https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2022/php/help.php</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2022/php/index.php">miRTarbase</a>.
<img src="/images/miRTarbase1.jpg?width=50pc" alt="miRTarbase1" /></p></li>
</ul>
<hr />
</div>
<div id="mir2disease" class="section level2">
<h2>3.3.6 <strong>miR2Disease</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: A manually curated database, aims at providing a comprehensive resource of miRNA deregulation in various human diseases. Each entry in the miR2Disease contains detailed information on a miRNA-disease relationship, including miRNA ID, disease name, a brief description of the miRNA-disease relationship, miRNA expression pattern in the disease state, detection method for miRNA expression, experimentally verified miRNA target gene(s), and literature reference .</p></li>
<li><p>Number of miRNAs: 349</p></li>
<li><p>Number of diseases: 163</p></li>
<li><p>Number of entries: 3273</p></li>
<li><p><strong>Web server</strong>: <a href="http://www.mir2disease.org/">miR2Disease</a>.</p></li>
</ul>
<hr />
</div>
<div id="tarbase" class="section level2">
<h2>3.3.7 <strong>TarBase</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: Computational predictive models are a key element of current systems biology. The focus of the DIANA lab is on the development of algorithms, databases and tools for interpreting and archiving genomic data in the framework of a systemic analysis. Current emphasis is on the analysis of microRNA (miRNA) and protein coding genes. MiRNAs are recently identified to be very abundant in mammalian organisms and play a key role in regulating development. Comprehensive models work and integrate data at various levels of biological detail. Therefore the activities of the DIANA lab range from the analysis of expression regulation from deep sequencing data, the annotation of miRNA regulatory elements and targets to the interpretation of the role of miRNAs in various diseases.
<img src="/images/TarBase.jpg?width=100pc" alt="TarBase" /></p></li>
<li><p><strong>Application</strong>: The TarBase tutorial can be found at
<a href="https://dianalab.e-ce.uth.gr/html/diana/web/index.php?r=tarbasev8%2Fhelp" class="uri">https://dianalab.e-ce.uth.gr/html/diana/web/index.php?r=tarbasev8%2Fhelp</a>.</p></li>
<li><p><strong>Web server</strong>: <a href="https://dianalab.e-ce.uth.gr/html/diana/web/index.php?r=tarbasev8">TarBase</a>.
<img src="/images/TarBase1.jpg?width=100pc" alt="TarBase1" /></p></li>
</ul>
<hr />
</div>
<div id="mirdip" class="section level2">
<h2>3.3.8 <strong>mirDIP</strong></h2>
<ul>
<li><p><strong>Introduction</strong>: mirDIP provide nearly 152 million human microRNA–target predictions, which were collected across 30 different resources. <img src="/images/mirDIP1.png?width=50pc" alt="mirDIP1" /></p></li>
<li><p><strong>Installation</strong>: You can use API for calling mirDIP from <a href="http://ophid.utoronto.ca/mirDIP/api_R.jsp">R</a>, <a href="http://ophid.utoronto.ca/mirDIP/api_Java.jsp">JAVA</a> and <a href="http://ophid.utoronto.ca/mirDIP/api_Python.jsp">Python</a> code.</p></li>
<li><p><strong>Web server</strong>: <a href="http://ophid.utoronto.ca/mirDIP/">mirDIP</a>.
<img src="/images/mirDIP.jpg?width=50pc" alt="mirDIP" /></p></li>
</ul>
</div>
