---
title: "TF gene interaction database"
output:
  blogdown::html_page:
    toc: true
weight: 2
---


<div id="TOC">
<ul>
<li><a href="#jaspar" id="toc-jaspar">2.2.1 <strong>JASPAR</strong></a></li>
<li><a href="#animaltfdb" id="toc-animaltfdb">2.2.2 <strong>AnimalTFDB</strong></a></li>
<li><a href="#chea" id="toc-chea">2.2.3 <strong>CHEA</strong></a></li>
<li><a href="#regnetwork" id="toc-regnetwork">2.2.4 <strong>RegNetwork</strong></a></li>
</ul>
</div>

<p>Transcription factors (TFs) are key regulators of intrinsic cellular processes, such as differentiation and development, and of the cellular response to external perturbation through signaling pathways. Most TFs recognize and bind to specific DNA sequences named as transcription factor biding sites, leading to specific spatiotemporal expression patterns of target genes.</p>
<p><strong>Breaking</strong>: Four TF target specific resources are collected.</p>
<hr />
<div id="jaspar" class="section level2">
<h2>2.2.1 <strong>JASPAR</strong></h2>
<p><strong>Introduction</strong>: JASPAR is a regularly maintained open-access database storing manually curated transcription factors (TF) binding profiles as position frequency matrices (PFMs). PFMs summarize occurrences of each nucleotide at each position in a set of observed TF-DNA interactions. PFMs can be transformed to probabilistic or energistic models to construct position weight matrices (PWMs) or position-specific scoring matrices (PSSMs), which can be used to scan any DNA sequence to predict TF binding sites (TFBSs). The JASPAR database provides TFBSs predicted using the profiles in the CORE collection.
<img src="/images/JASPAR1.jpg?width=50pc" alt="JASPAR1" /></p>
<p><strong>Installation</strong>: Different releases of JASPAR can also be accessed through Bioconductor data packages. Currently four JASPAR releases are available, and to install those packages, start R (version “4.2”) and enter:
+ JASPAR2020:</p>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;JASPAR2020&quot;)</code></pre>
<ul>
<li>JASPAR2018</li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;JASPAR2018&quot;)</code></pre>
<ul>
<li>JASPAR2016</li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;JASPAR2016&quot;)</code></pre>
<ul>
<li>JASPAR2014</li>
</ul>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;JASPAR2014&quot;)</code></pre>
<p><strong>Application</strong>: The JASPAR tutorial can be found at<a href="https://jaspar.genereg.net/docs/" class="uri">https://jaspar.genereg.net/docs/</a>.
The JASPAR2020 vignette can be found at <a href="http://bioconductor.org/packages/release/data/annotation/vignettes/JASPAR2020/inst/doc/JASPAR2020.html" class="uri">http://bioconductor.org/packages/release/data/annotation/vignettes/JASPAR2020/inst/doc/JASPAR2020.html</a>.
The JASPAR2018 vignette can be found at <a href="http://bioconductor.org/packages/release/data/annotation/manuals/JASPAR2018/man/JASPAR2018.pdf" class="uri">http://bioconductor.org/packages/release/data/annotation/manuals/JASPAR2018/man/JASPAR2018.pdf</a>.
The JASPAR2016 vignette can be found at <a href="http://bioconductor.org/packages/release/data/experiment/manuals/JASPAR2016/man/JASPAR2016.pdf" class="uri">http://bioconductor.org/packages/release/data/experiment/manuals/JASPAR2016/man/JASPAR2016.pdf</a>.
The JASPAR2016 vignette can be found at &lt;<a href="http://bioconductor.org/packages/release/data/experiment/manuals/JASPAR2014/man/JASPAR2014.pdf" class="uri">http://bioconductor.org/packages/release/data/experiment/manuals/JASPAR2014/man/JASPAR2014.pdf</a>.</p>
<p><strong>Web server</strong>: <a href="https://jaspar.genereg.net/">JASPAR</a>.
<img src="/images/JASPAR.jpg?width=50pc" alt="JASPAR" /></p>
<hr />
</div>
<div id="animaltfdb" class="section level2">
<h2>2.2.2 <strong>AnimalTFDB</strong></h2>
<p><strong>Introduction</strong>: AnimalTFDB is a comprehensive database including classification and annotation of genome-wide transcription factors (TFs) and transcription cofactors in 183 animal genomes. In AnimalTFDB v4.0, there are total 270k TFs which are further classified into 73 families and six categories based on their DNA-binding domain (DBD). Meanwhile, there are also 150k cofactors classified into 82 families and six categories.</p>
<center>
<b>Database information</b>
</center>
<table>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Species</td>
<td align="left">183</td>
</tr>
<tr class="even">
<td align="left">TFs</td>
<td align="left">2700000</td>
</tr>
<tr class="odd">
<td align="left">TF Cofactors</td>
<td align="left">1500000</td>
</tr>
<tr class="even">
<td align="left">TF Families</td>
<td align="left">73</td>
</tr>
<tr class="odd">
<td align="left">TF Cofactor Families</td>
<td align="left">83</td>
</tr>
</tbody>
</table>
<p><img src="/images/AnimalTFDB.png?width=30pc" alt="AnimalTFDB" />
<strong>Application</strong>: The AnimalTFDB tutorial can be found at<a href="http://bioinfo.life.hust.edu.cn/AnimalTFDB4/static/site/" class="uri">http://bioinfo.life.hust.edu.cn/AnimalTFDB4/static/site/</a>.</p>
<p><strong>Web server</strong>: <a href="http://bioinfo.life.hust.edu.cn/AnimalTFDB4/#/">AnimalTFDB</a>.
<img src="/images/AnimalTFDB1.jpg?width=50pc" alt="AnimalTFDB1" /></p>
<hr />
</div>
<div id="chea" class="section level2">
<h2>2.2.3 <strong>CHEA</strong></h2>
<p>link to SigCom LINCS</p>
<hr />
</div>
<div id="regnetwork" class="section level2">
<h2>2.2.4 <strong>RegNetwork</strong></h2>
<p><strong>Introduction</strong>: RegNetwork is a database of transcriptional and posttranscriptional regulatory networks in human and mouse. TF and miRNA are two major regulators controlling gene expression. RegNetwork collects the knowledge-based regulatory relationships, as well as some potentially regulatory relationships between the two regulators and targets. It provides a platform of depositing the known and predicted gene regulations in the transcriptional and posttranscriptional levels simultaneously. The knowledge-derived regulatory networks is expected to be greatly beneficial for identifying critical regulatory programs in various context-specific conditions.
<img src="/images/RegNetwork1.jpg?width=50pc" alt="RegNetwork1" /></p>
<p><strong>Application</strong>: The AnimalTFDB tutorial can be found at <a href="https://regnetworkweb.org/about.jsp" class="uri">https://regnetworkweb.org/about.jsp</a>.</p>
<p><strong>Web server</strong>: <a href="https://regnetworkweb.org/">RegNetwork</a>.
<img src="/images/RegNetwork.jpg?width=70pc" alt="RegNetwork" /></p>
</div>
